```{r}
# The aim of this part is to calculate the number of reads for each TE family, by summing reads of each locus by TE family

library(dplyr)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/")
#load file
data_194MR=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_194MR_mobilome.paf.merged200_rd_all_194MR_mobilome_mobilome_minimap_tmp.paf.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_194MR[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))

#select relevant columns
data_rc=unique(data_194MR[,c(1:4,19)])
#sum reads for each TE family
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))

data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clust194MR_all=data[,c(1,2)]



data_216MR=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_216MR_mobilome.paf.merged200_rd_all_216MR_mobilome_mobilome_minimap_tmp.paf.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_216MR[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(data_216MR[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clust216MR_all=data[,c(1,2)]

data_BA20=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_BA20-001_mobilome_minimap.paf.merged200_rd_all_BA20-001_mobilome_minimap_tmp.paf.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_BA20[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(data_BA20[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clustBA20_all=data[,c(1,2)]

data_V25=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_V25-001_mobilome_minimap.paf.merged200_rd_all_V25-001_mobilome_minimap_tmp.paf.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_V25[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(data_V25[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clustV25_all=data[,c(1,2)]

data_CC9=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_CC9_mobilome.paf.merged200_rd_all_CC9_mobilome_mobilome_minimap_tmp.paf.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_CC9[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(data_CC9[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clustCC9_all=data[,c(1,2)]

data_BD18=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_BD18.q8-v3.paf.merged200_rd_all_BD18.q8_mobilome_minimap.paf_filt.bed.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_BD18[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(data_BD18[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clustBD18_all=data[,c(1,2)]

data_BV15=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/all_BV15_mobilome_minimap.paf.merged200_rd_all_BV15_mobilome_minimap_tmp.paf.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(data_BV15[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(data_BV15[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
clustBV15_all=data[,c(1,2)]


barcode01=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode01/barcode01_mobilome.paf.merged200_rd_barcode01_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode01[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode01[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode01_all=data[,c(1,2)]


barcode02=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode02/barcode02.paf.merged200_rd_barcode02_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode02[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode02[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode02_all=data[,c(1,2)]

barcode03=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode03/barcode03.paf.merged200_rd_barcode03_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode03[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode03[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode03_all=data[,c(1,2)]

barcode04=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode04/barcode04.paf.merged200_rd_barcode04_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode04[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode04[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode04_all=data[,c(1,2)]


barcode05=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode05/barcode05.paf.merged200_rd_barcode05_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode05[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode05[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode05_all=data[,c(1,2)]

barcode06=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode06/barcode06.paf.merged200_rd_barcode06_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode06[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode06[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode06_all=data[,c(1,2)]


barcode07=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode07/barcode07.paf.merged200_rd_barcode07_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode07[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode07[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100 ) ,]
barcode07_all=data[,c(1,2)]

barcode08=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode08/barcode08.paf.merged200_rd_barcode08_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode08[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode08[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode08_all=data[,c(1,2)]


barcode09=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode09/barcode09.paf.merged200_rd_barcode09_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode09[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode09[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode09_all=data[,c(1,2)]

barcode10=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode10/barcode10.paf.merged200_rd_barcode10_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode10[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode10[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode10_all=data[,c(1,2)]

barcode11=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode11/barcode11.paf.merged200_rd_barcode11_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode11[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode11[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode11_all=data[,c(1,2)]

barcode12=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode12/barcode12.paf.merged200_rd_barcode12_mobilome.paf.filt.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode12[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode12[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode12_all=data[,c(1,2)]

barcode20=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode20/all_barcode20.paf.merged200_rd_all_barcode20.paf_filt.bed.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode20[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode20[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode20_all=data[,c(1,2)]

barcode21=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode21/all_barcode21.paf.merged200_rd_all_barcode21.paf_filt.bed.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode21[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode21[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode21_all=data[,c(1,2)]

barcode22=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode22/all_barcode22.paf.merged200_rd_all_barcode22.paf_filt.bed.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode22[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode22[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode22_all=data[,c(1,2)]

barcode23=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode23/all_barcode23.paf.merged200_rd_all_barcode23.paf_filt.bed.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode23[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode23[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode23_all=data[,c(1,2)]

barcode24=read.table(file="/media/lgdp/Seagate Basic/P2/Mobilome_LR_2024/barcode24/all_barcode24.paf.merged200_rd_all_barcode24.paf_filt.bed.py_out3_loj-vs-TEs.bed",sep="\t")
data_rep=unique(barcode24[,c(8,19)])
data_rep=data_rep[data_rep$V8!=".",]
data_rep=data_rep %>% group_by(V19) %>% summarise(rep = n_distinct(V8))
data_rc=unique(barcode24[,c(1:4,19)])
data_rc=data_rc%>% group_by(V19) %>% summarise(rd = sum(V4))
data=left_join(data_rc,data_rep,by=c("V19"="V19"))
data=data[(( data$rd>=5 && !is.na(data$rep) && data$rep>=1) || data$rd>=100) ,]
barcode24_all=data[,c(1,2)]


```


```{r}
# The part of this script is to use new TE cluster names replacing TE families used in older annotation for all mobilome sequencing data and concatenate the results

library(stringr)
library(tibble)
library(ggplot2)
library(pheatmap)
library(openxlsx)

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/guppy_hac_fastq_FAS09199_R9")

# load file containing equivalence of old TE families and new TE clusters
cluster=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
# load file containing length of consensus sequence of old TE families
size=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_size.txt")

# concatenate results old TE family names found in mobilome samples
te_fam=unique(rbind(clust194MR_all$V19,clust216MR_all$V19,clustBA20_all$V19,clustV25_all$V19,clustCC9_all$V19,clustBD18_all$V19,clustBV15_all$V19,barcode01_all$V19,barcode02_all$V19,barcode03_all$V19,barcode04_all$V19,barcode05_all$V19,barcode06_all$V19,barcode07_all$V19,barcode08_all$V19,barcode09_all$V19,barcode10_all$V19,barcode11_all$V19,barcode12_all$V19,barcode20_all$V19,barcode21_all$V19,barcode22_all$V19,barcode23_all$V19,barcode24_all$V19))
# Get the TE classification
family=unique(data.frame(str_split_fixed(te_fam, "_", 4)))

family$X1=str_replace_all(family$X1,"MITE|MITE","MITE")
family$X1=str_replace_all(family$X1,"\\|NA","")
family$X2=str_replace_all(family$X2,"MITE|MITE","MITE")
family$X2=str_replace_all(family$X2,"\\|NA","")

# Join new cluster name of old TE family and retaining TE classification
te_fam=left_join(family,cluster,by=c("X4"="V1"))
# Add TE classification to Cluster ID
te_fam$name=paste0(te_fam$X1,"_",te_fam$V2)
te_fam$id=paste0(te_fam$X1,"_",te_fam$X2,"_",te_fam$X3,"_",te_fam$X4)
# Join mean size of TE family
te_fam=left_join(te_fam,size,by=c("X4"="V1"))

copies=read.xlsx("/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_TE_stats_classif.xlsx")
copies=copies[,c(1,6)]
te_fam=left_join(te_fam,copies,by=c("X4"="Seq_name.Cur4"))
te_fam=unique(te_fam[,c(6:9)])



# Join the read number for each cluster in all mobilome samples
data=left_join(te_fam,clust194MR_all,by=c("id"="V19"))
data=left_join(data,clust216MR_all,by=c("id"="V19"))
data=left_join(data,clustBA20_all,by=c("id"="V19"))
data=left_join(data,clustV25_all,by=c("id"="V19"))
data=left_join(data,clustCC9_all,by=c("id"="V19"))
data=left_join(data,clustBD18_all,by=c("id"="V19"))
data=left_join(data,clustBV15_all,by=c("id"="V19"))
data=left_join(data,barcode01_all,by=c("id"="V19"))
data=left_join(data,barcode02_all,by=c("id"="V19"))
data=left_join(data,barcode03_all,by=c("id"="V19"))
data=left_join(data,barcode04_all,by=c("id"="V19"))
data=left_join(data,barcode05_all,by=c("id"="V19"))
data=left_join(data,barcode06_all,by=c("id"="V19"))
data=left_join(data,barcode07_all,by=c("id"="V19"))
data=left_join(data,barcode08_all,by=c("id"="V19"))
data=left_join(data,barcode09_all,by=c("id"="V19"))
data=left_join(data,barcode10_all,by=c("id"="V19"))
data=left_join(data,barcode11_all,by=c("id"="V19"))
data=left_join(data,barcode12_all,by=c("id"="V19"))
data=left_join(data,barcode20_all,by=c("id"="V19"))
data=left_join(data,barcode21_all,by=c("id"="V19"))
data=left_join(data,barcode22_all,by=c("id"="V19"))
data=left_join(data,barcode23_all,by=c("id"="V19"))
data=left_join(data,barcode24_all,by=c("id"="V19"))


# Replace NA by 0
data[is.na(data)] <- 0
data=data[,-c(2)]

# Rename columns
# Data frame with the following column : TE cluster name, size of old TE family, samples 
colnames(data)=c("name","size","copies","v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","barcode01","barcode02","barcode03","barcode04","barcode05","barcode06","barcode07","barcode08","barcode09","barcode10","barcode11","barcode12","barcode20","barcode21","barcode22","barcode23","barcode24")
data_size=data[,c(1,2)]
data_copies=data[,c(1,3)]
data=data[,-c(2,3)]

```


```{r}
# The part of this script is normalize reads of each TE family by mean size of TE consensus and library size, use data to plot an heatmap

# A cluster can have multiple TE families so multiple size
# Mean size of each cluster is calculated from size of old TE families

data=data %>% group_by(name) %>% summarise(across(everything(), sum))
data_size=data_size %>% group_by(name) %>% summarise_each(funs(mean)) 
data_copies=data_copies%>% group_by(name) %>% summarise(nb = mean(copies)) 
data=left_join(data,data_size,by=c("name"="name"))
data=left_join(data,data_copies,by=c("name"="name"))

#data=rbind(data,mito_chloro1)


rd_data_lr=data
data=rd_data_lr
write.csv(data,file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/mobilome_LR_samples_raw_counts.csv",sep="\t",row.names=F,quote=F)
# Normalizing read number of TE cluster with mean size of sequence of TE cluster for each sample
data$v_194MR=data$v_194MR/data$size
data$v_216MR=data$v_216MR/data$size
data$BA20=data$BA20/data$size
data$V25=data$V25/data$size
data$CC9=data$CC9/data$size
data$BD18=data$BD18/data$size
data$BV15=data$BV15/data$size
data$barcode01=data$barcode01/data$size
data$barcode02=data$barcode02/data$size
data$barcode03=data$barcode03/data$size
data$barcode04=data$barcode04/data$size
data$barcode05=data$barcode05/data$size
data$barcode06=data$barcode06/data$size
data$barcode07=data$barcode07/data$size
data$barcode08=data$barcode08/data$size
data$barcode09=data$barcode09/data$size
data$barcode10=data$barcode10/data$size
data$barcode11=data$barcode11/data$size
data$barcode12=data$barcode12/data$size
data$barcode20=data$barcode20/data$size
data$barcode21=data$barcode21/data$size
data$barcode22=data$barcode22/data$size
data$barcode23=data$barcode23/data$size
data$barcode24=data$barcode24/data$size

# Normalizing read number of TE cluster with library size for each sample
data$v_194MR=data$v_194MR/0.215872
data$v_216MR=data$v_216MR/0.190463
data$BA20=data$BA20/0.138765
data$V25=data$V25/0.611088
data$CC9=data$CC9/0.793042
data$BD18=data$BD18/0.695400
data$BV15=data$BV15/0.022979
data$barcode01=data$barcode01/0.759311
data$barcode02=data$barcode02/0.599277
data$barcode03=data$barcode03/0.915870
data$barcode04=data$barcode04/0.558506
data$barcode05=data$barcode05/0.372519
data$barcode06=data$barcode06/0.715171
data$barcode07=data$barcode07/1.159134
data$barcode08=data$barcode08/0.601451
data$barcode09=data$barcode09/1.081948
data$barcode10=data$barcode10/0.438857
data$barcode11=data$barcode11/0.272462
data$barcode12=data$barcode12/0.053927
data$barcode20=data$barcode20/0.233342
data$barcode21=data$barcode21/0.392087
data$barcode22=data$barcode22/0.410388
data$barcode23=data$barcode23/0.410388
data$barcode24=data$barcode24/0.323116

# Remove column with size
data=data[,-c(20,26,27)]
# Remove unclassified columns
data=data[data$name!="LTR_NA",]
write.csv(data,file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/mobilome_LR_samples_RPKM.csv",sep="\t",row.names=F,quote=F)
# Rename rownames as TE clusters
data=data.frame(column_to_rownames(data, var = "name"))
data2=data
data=data2


data=data[data$v_194MR>0 | data$v_216MR>0 | data$BA20>0 | data$BD18>0 | data$CC9>0 | data$V25>0 | data$BV15>0 | data$barcode01>0 | data$barcode02>0 | data$barcode03>0 | data$barcode04>0 | data$barcode05>0 | data$barcode06>0 | data$barcode07>0 | data$barcode08>0 | data$barcode09>0 | data$barcode10>0 | data$barcode11>0  | data$barcode20>0 | data$barcode21>0 | data$barcode22>0  | data$barcode23>0 | data$barcode24>0 ,]
data_beforescale=data
#data=data[,-c(18)]
# Scaling columns for heatmap
data$v_194MR=scale(data$v_194MR)
data$v_216MR=scale(data$v_216MR)
data$BA20=scale(data$BA20)
data$CC9=scale(data$CC9)
data$V25=scale(data$V25)
data$BD18=scale(data$BD18)
data$BV15=scale(data$BV15)
data$barcode01=scale(data$barcode01)
data$barcode02=scale(data$barcode02)
data$barcode03=scale(data$barcode03)
data$barcode04=scale(data$barcode04)
data$barcode05=scale(data$barcode05)
data$barcode06=scale(data$barcode06)
data$barcode07=scale(data$barcode07)
data$barcode08=scale(data$barcode08)
data$barcode09=scale(data$barcode09)
data$barcode10=scale(data$barcode10)
data$barcode11=scale(data$barcode11)
#data$barcode12=scale(data$barcode12)
data$barcode20=scale(data$barcode20)
data$barcode21=scale(data$barcode21)
data$barcode22=scale(data$barcode22)
data$barcode23=scale(data$barcode23)
data$barcode24=scale(data$barcode24)

data=data[,-c(10)]
#colnames(data)=c("194MR_bud","216MR_bud","BA20_leaf","CC9_leaf","V25_leaf","BD18_leaf","BV15_leaf","185M_bud","354M_bud","CBRR_bud","525M_bud","525R_bud","GE1_leaf","BD18_leaf_2","BV15_leaf_2","CC9_leaf_2","V25_leaf_2","216M_leaf")
colnames(data)=c("194MR_bud","216MR_bud","BA20_leaf","CC9_leaf","V25_leaf","BD18_leaf","BV15_leaf","185M_bud","354M_bud","525M_bud","525R_bud","GE1_leaf","BD18_leaf_2","BV15_leaf_2","CC9_leaf_2","V25_leaf_2","216M_leaf","CC9_bud","H8007_bud","BA20_bud","BD18_bud","V25_bud")
data=data[,-which(names(data) %in% c("BD18_leaf_2","BV15_leaf_2","CC9_leaf_2","V25_leaf_2"))]



#data=data[,-c(8:17)]
#data=data[,-c(9,11)]
# converting data frame into matrix

mtscaled_lr <- as.matrix(data)

# Renaming columns 
#colnames(mtscaled_lr)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","barcode01","barcode02","barcode04","barcode05","barcode06","barcode07","barcode08","barcode09","barcode10","barcode11","barcode12")
# Sorting rows by rownames
mtscaled_lr<-mtscaled_lr[order(rownames(mtscaled_lr)),]
write.table(rownames(data),file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/TE_eccDNA_LR.txt",quote=F,row.names = F,col.names = F)
# Selecting colors for heatmap
#hmcols<-colorRampPalette(c("grey92","#de3e51", "#962483"))(round((max(data)),digit=0))
hmcols<-colorRampPalette(c("white","mediumorchid3", "mediumpurple4"))(round((max(data)),digit=0))

# Ploting heatmap
library(grid)
ph=pheatmap(mtscaled_lr,cluster_rows = T,rot = 45,cluster_cols = T, fontsize_row =0.8, fontsize_col = 15, margins=c(7,3),color=hmcols,legend_labels =T)
ph$gtable$grobs[[1]]$gp <- gpar(lwd = 0)
ph$gtable$grobs[[2]]$gp <- gpar(col = 'blue')
grid.newpage()
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("heatmap_194_216_BA2_V25_CC9_BD18_BV15_RC_barcode_q8_nosize.png",res=900,width=8000,height = 16000)
grid.draw(ph$gtable)
dev.off()

pheatmap_result <- pheatmap(mtscaled_lr,cluster_rows = T,rot = 45,cluster_cols = T, fontsize_row =1, fontsize_col = 15, margins=c(7,3),color=hmcols,legend_labels =T)

clustered_rows <- pheatmap_result$tree_row$order
selected_rows1 <- clustered_rows[0:50]
selected_rows2 <- clustered_rows[1640:1690]
selected_rows=c(selected_rows1,selected_rows2)
subset_scaled_data <- data[selected_rows, ]
clustered_cols <- pheatmap_result$tree_col$order
subset_scaled_data <- subset_scaled_data %>% select(all_of(clustered_cols))

ph=pheatmap(subset_scaled_data,cluster_rows = F,rot = 45,cluster_cols = F,fontsize_row =13,margins=c(1,1),color=hmcols)
ph$gtable$grobs[[1]]$gp <- gpar(lwd = 0)
ph$gtable$grobs[[2]]$gp <- gpar(col = 'blue')

library(grid)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("heatmap_194_216_BA2_V25_CC9_BD18_BV15_RC_barcode_q8_nosize_1.png",res=900,width=8000,height = 16000)
grid.newpage()
grid.draw(ph$gtable)
dev.off()


clustered_rows <- pheatmap_result$tree_row$order
selected_rows1 <- clustered_rows[0:50]
selected_rows2 <- clustered_rows[1640:1690]
selected_rows=c(selected_rows1,selected_rows2)
subset_scaled_data <- unique(data[selected_rows, ])
clustered_cols <- pheatmap_result$tree_col$order
subset_scaled_data <- subset_scaled_data %>% select(all_of(clustered_cols))
#subset_scaled_data=subset(subset_scaled_data, select=-c(CC9_leaf_2,BD18_leaf_2,BV15_leaf_2,V25_leaf_2))
ph=pheatmap(subset_scaled_data,cluster_rows = F,rot = 45,cluster_cols = F,margins=c(1,1),color=hmcols,fontsize_row =13)
ph$gtable$grobs[[1]]$gp <- gpar(lwd = 0)
ph$gtable$grobs[[2]]$gp <- gpar(col = 'blue')

library(grid)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("heatmap_194_216_BA2_V25_CC9_BD18_BV15_RC_barcode_q8_1_filt.png",res=900,width=8000,height = 16000)
grid.newpage()
grid.draw(ph$gtable)
dev.off()


data_copies=left_join(te_fam,data_copies,by=c("name"="name"))
write.csv(mtscaled_lr,file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/matrix_heatmap.csv",row.names = T,col.names = T,quote = F)
write.csv(data_copies,file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/data_copies.csv",row.names = T,col.names = T,quote = F)


te_fam=data_copies[data_copies$name %in% rownames(subset_scaled_data),]
te_fam1=data_copies%>% group_by(name) %>% summarise(nb = mean(copies)) 

te_fam<- te_fam1 %>% column_to_rownames(var = "name")
rows=rownames(subset_scaled_data)
row_order <- match(rows, rownames(te_fam))
te_fam <- te_fam[row_order,, drop = FALSE ]
rownames(te_fam) <- rows
te_fam$nb=scale(te_fam$nb)
hmcols<-colorRampPalette(c("white","mediumorchid3", "mediumpurple4"))(round((max(data)),digit=0))


png("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/heatmap_copynumber_TE_family_filt5.png",res=900,width=8000,height = 14000)
pheatmap(te_fam,cluster_rows = F,rot = 45,cluster_cols = F, fontsize_row =5, fontsize_col = 15, margins=c(7,3),color=hmcols,legend_labels =T,show_rownames=TRUE)

```


```{r}
# The part of this script is normalize reads of each TE family by mean size of TE consensus and library size, use data to plot an heatmap

# A cluster can have multiple TE families so multiple size
# Mean size of each cluster is calculated from size of old TE families

data=data %>% group_by(name) %>% summarise(across(everything(), sum))
data_size=data_size %>% group_by(name) %>% summarise_each(funs(mean)) 
data_copies=data_copies%>% group_by(name) %>% summarise(nb = mean(copies)) 
data=left_join(data,data_size,by=c("name"="name"))
data=left_join(data,data_copies,by=c("name"="name"))

#data=rbind(data,mito_chloro1)


rd_data_lr=data

# Normalizing read number of TE cluster with mean size of sequence of TE cluster for each sample
data$v_194MR=data$v_194MR/data$size
data$v_216MR=data$v_216MR/data$size
data$BA20=data$BA20/data$size
data$V25=data$V25/data$size
data$CC9=data$CC9/data$size
data$BD18=data$BD18/data$size
data$BV15=data$BV15/data$size
data$barcode01=data$barcode01/data$size
data$barcode02=data$barcode02/data$size
data$barcode04=data$barcode04/data$size
data$barcode05=data$barcode05/data$size
data$barcode06=data$barcode06/data$size
data$barcode07=data$barcode07/data$size
data$barcode08=data$barcode08/data$size
data$barcode09=data$barcode09/data$size
data$barcode10=data$barcode10/data$size
data$barcode11=data$barcode11/data$size
data$barcode12=data$barcode12/data$size

# Normalizing read number of TE cluster with copy number of TE cluster for each sample
data$v_194MR=data$v_194MR/data$nb
data$v_216MR=data$v_216MR/data$nb
data$BA20=data$BA20/data$nb
data$V25=data$V25/data$nb
data$CC9=data$CC9/data$nb
data$BD18=data$BD18/data$nb
data$BV15=data$BV15/data$nb
data$barcode01=data$barcode01/data$nb
data$barcode02=data$barcode02/data$nb
data$barcode04=data$barcode04/data$nb
data$barcode05=data$barcode05/data$nb
data$barcode06=data$barcode06/data$nb
data$barcode07=data$barcode07/data$nb
data$barcode08=data$barcode08/data$nb
data$barcode09=data$barcode09/data$nb
data$barcode10=data$barcode10/data$nb
data$barcode11=data$barcode11/data$nb
#data$barcode12=data$barcode12/data$nb

# Normalizing read number of TE cluster with library size for each sample
data$v_194MR=data$v_194MR/0.215872
data$v_216MR=data$v_216MR/0.190463
data$BA20=data$BA20/0.138765
data$V25=data$V25/0.611088
data$CC9=data$CC9/0.793042
data$BD18=data$BD18/0.695400
data$BV15=data$BV15/0.022979
data$barcode01=data$barcode01/0.759311
data$barcode02=data$barcode02/0.599277
data$barcode04=data$barcode04/0.558506
data$barcode05=data$barcode05/0.372519
data$barcode06=data$barcode06/0.715171
data$barcode07=data$barcode07/1.159134
data$barcode08=data$barcode08/0.601451
data$barcode09=data$barcode09/1.081948
data$barcode10=data$barcode10/0.438857
data$barcode11=data$barcode11/0.272462
data$barcode12=data$barcode12/0.053927

# Remove column with size
data=data[,-c(20,21)]
# Remove unclassified columns
data=data[data$name!="LTR_NA",]
# Rename rownames as TE clusters
data=data.frame(column_to_rownames(data, var = "name"))
data=data[data$v_194MR>0 | data$v_216MR>0 | data$BA20>0 | data$BD18>0 | data$CC9>0 | data$V25>0 | data$BV15>0 | data$barcode01>0 | data$barcode02>0 | data$barcode04>0 | data$barcode05>0 | data$barcode06>0 | data$barcode07>0 | data$barcode08>0 | data$barcode09>0 | data$barcode10>0 | data$barcode11>0 | data$barcode12>0 ,]
data_beforescale=data

# Scaling columns for heatmap
data$v_194MR=scale(data$v_194MR)
data$v_216MR=scale(data$v_216MR)
data$BA20=scale(data$BA20)
data$CC9=scale(data$CC9)
data$V25=scale(data$V25)
data$BD18=scale(data$BD18)
data$BV15=scale(data$BV15)
data$barcode01=scale(data$barcode01)
data$barcode02=scale(data$barcode02)
data$barcode04=scale(data$barcode04)
data$barcode05=scale(data$barcode05)
data$barcode06=scale(data$barcode06)
data$barcode07=scale(data$barcode07)
data$barcode08=scale(data$barcode08)
data$barcode09=scale(data$barcode09)
data$barcode10=scale(data$barcode10)
data$barcode11=scale(data$barcode11)
data$barcode12=scale(data$barcode12)

data=data[,-c(18)]
#data=data[,-c(9,11)]
# converting data frame into matrix
mtscaled_lr <- as.matrix(data)

# Renaming columns 
#colnames(mtscaled_lr)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","barcode01","barcode02","barcode04","barcode05","barcode06","barcode07","barcode08","barcode09","barcode10","barcode11","barcode12")
# Sorting rows by rownames
mtscaled_lr<-mtscaled_lr[order(rownames(mtscaled_lr)),]

# Selecting colors for heatmap
hmcols<-colorRampPalette(c("white","#de3e51", "#962483"))(round((max(data)),digit=0))

# Ploting heatmap
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("heatmap_194_216_BA2_V25_CC9_BD18_BV15_RC_q8.png",res=900,width=7000,height = 19000)
pheatmap(mtscaled_lr,cluster_rows = T,rot = 45,cluster_cols = T, fontsize_row =1, fontsize_col = 10, margins=c(7,3),color=hmcols,legend_labels =T)
```


```{r}
data=rd_data_lr
#data=data[,-c(20,21)]
data$row_sum_leaf <- rowSums(data[ , c(4:8,13,15:18)], na.rm=TRUE)
data$row_sum_bud <- rowSums(data[ , c(2,3,9:12,14)], na.rm=TRUE)
data1=data[,c(1,20,22,23)]
data1=data1[data1$name!="LTR_NA",]
nb1=215872+190463+759311+599277+558506+372519+1159134
nb2=138765+611088+793042+695400+22979+715171+601451+1081948+438857+272462

data1$row_sum_leaf=data1$row_sum_leaf/data1$size
data1$row_sum_bud=data1$row_sum_bud/data1$size
data1$row_sum_leaf=data1$row_sum_leaf/nb2
data1$row_sum_bud=data1$row_sum_bud/nb1
data1=data1[,-c(2,4)]
data1=data.frame(column_to_rownames(data1, var = "name"))

mtscaled_lr <- as.matrix(data1)
mtscaled_lr<-mtscaled_lr[order(rownames(mtscaled_lr)),]
names=names=(mtscaled_lr[mtscaled_lr>=1.000639e-08])
mtscaled_lr=mtscaled_lr[mtscaled_lr>=1.000639e-08]
names(mtscaled_lr)=names
# Selecting colors for heatmap
hmcols<-colorRampPalette(c("white","#de3e51", "#962483"))(50)

# Ploting heatmap
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("heatmap_194_216_BA2_V25_CC9_BD18_BV15_RC_q8_pool.png",res=900,width=7000,height = 19000)
pheatmap(mtscaled_lr,cluster_rows = T,rot = 45,cluster_cols = F, fontsize_row =10, fontsize_col = 15, margins=c(7,3),color=hmcols,legend_labels =T)

```


```{r}
#The part of this script is to calculate top families by individual;, count the occurence in top of these TE families and plot an heatmap
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/")
data=data_beforescale

# Sorting TE cluster by read numbers and selecting 
r1=rownames(data)[order(data$v_194MR, decreasing=TRUE)][1:5]
# Convert from list of TE clusters to dataframe with a single column
r1=data.frame(V4.x=c(t(r1)))
# Add a 2nd column with sample name
r1$ind="v_194MR"

r2=rownames(data)[order(data$v_216MR, decreasing=TRUE)][1:5]
r2=data.frame(V4.x=c(t(r2)))
r2$ind="v_216MR"

r3=rownames(data)[order(data$BA20, decreasing=TRUE)][1:5]
r3=data.frame(V4.x=c(t(r3)))
r3$ind="BA20"

r4=rownames(data)[order(data$V25, decreasing=TRUE)][1:5]
r4=data.frame(V4.x=c(t(r4)))
r4$ind="V25"

r5=rownames(data)[order(data$CC9, decreasing=TRUE)][1:5]
r5=data.frame(V4.x=c(t(r5)))
r5$ind="CC9"

r6=rownames(data)[order(data$BD18, decreasing=TRUE)][1:5]
r6=data.frame(V4.x=c(t(r6)))
r6$ind="BD18"

r7=rownames(data)[order(data$BV15, decreasing=TRUE)][1:5]
r7=data.frame(V4.x=c(t(r7)))
r7$ind="BV15"

r8=rownames(data)[order(data$barcode01, decreasing=TRUE)][1:5]
r8=data.frame(V4.x=c(t(r8)))
r8$ind="185M"

r9=rownames(data)[order(data$barcode02, decreasing=TRUE)][1:5]
r9=data.frame(V4.x=c(t(r9)))
r9$ind="354R"

r10=rownames(data)[order(data$barcode04, decreasing=TRUE)][1:5]
r10=data.frame(V4.x=c(t(r10)))
r10$ind="525M"

r11=rownames(data)[order(data$barcode05, decreasing=TRUE)][1:5]
r11=data.frame(V4.x=c(t(r11)))
r11$ind="525R"


r12=rownames(data)[order(data$barcode06, decreasing=TRUE)][1:5]
r12=data.frame(V4.x=c(t(r12)))
r12$ind="GE1"

r13=rownames(data)[order(data$barcode07, decreasing=TRUE)][1:5]
r13=data.frame(V4.x=c(t(r13)))
r13$ind="BD18"


r14=rownames(data)[order(data$barcode08, decreasing=TRUE)][1:5]
r14=data.frame(V4.x=c(t(r14)))
r14$ind="BV15"

r15=rownames(data)[order(data$barcode09, decreasing=TRUE)][1:5]
r15=data.frame(V4.x=c(t(r15)))
r15$ind="CC9"

r16=rownames(data)[order(data$barcode10, decreasing=TRUE)][1:5]
r16=data.frame(V4.x=c(t(r16)))
r16$ind="V25"

r17=rownames(data)[order(data$barcode11, decreasing=TRUE)][1:5]
r17=data.frame(V4.x=c(t(r17)))
r17$ind="216M"

r18=rownames(data)[order(data$barcode03, decreasing=TRUE)][1:5]
r18=data.frame(V4.x=c(t(r18)))
r18$ind="CBRR"

r19=rownames(data)[order(data$barcode20, decreasing=TRUE)][1:5]
r19=data.frame(V4.x=c(t(r19)))
r19$ind="BV15"

r20=rownames(data)[order(data$barcode21, decreasing=TRUE)][1:5]
r20=data.frame(V4.x=c(t(r20)))
r20$ind="CC9"

r21=rownames(data)[order(data$barcode22, decreasing=TRUE)][1:5]
r21=data.frame(V4.x=c(t(r21)))
r21$ind="H8_007"

r22=rownames(data)[order(data$barcode23, decreasing=TRUE)][1:5]
r22=data.frame(V4.x=c(t(r22)))
r22$ind="BA20"

r23=rownames(data)[order(data$barcode24, decreasing=TRUE)][1:5]
r23=data.frame(V4.x=c(t(r23)))
r23$ind="BD18"

r24=rownames(data)[order(data$barcode24, decreasing=TRUE)][1:5]
r24=data.frame(V4.x=c(t(r24)))
r24$ind="V25"


# Concatenate all samples top clusters data frame
top10=rbind(r1,r2,r3,r4,r5,r6,r7,r8,r9,r18,r10,r11,r12,r17,r19,r20,r21,r22,r23,r24)

# Count the occurance of each TE cluster to find in how many samples the TE cluster is present as most top/abundant family
top10=top10 %>% group_by(V4.x) %>% add_count(name = "nb_ind")
top10=unique(top10[,c(1,3)])
rownames = top10$V4.x

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/")
png("barplot_top10_all.png",res=800,width=7000,height = 6000)
ggplot(top10[top10$nb_ind>=2,],aes(x=reorder(V4.x,-nb_ind),y=nb_ind))+geom_bar(stat="identity",fill="black") +theme_classic() + theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=1))+ scale_y_continuous(limits = c(0, 12), breaks = seq(0, 12, by = 1)) + theme(axis.line=element_line(size=0.75)) 


write.table(top10[top10$inb_ind>=1,]$V4.x,file="listtop10.txt",quote=F,row.names = F,col.names = F)

top10=top10[,-c(1)]
rownames(top10)=rownames

#top10<-top10[order(rownames(top10)),]

hmcols<-colorRampPalette(c("white","orange1", "orange3"))(7)
rownames(top10[top10$inb_ind>=4,])
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("heatmap_top10.png",res=800,width=9000,height = 14000)
pheatmap(top10,cluster_rows = T,rot = 45,cluster_cols = F, fontsize_row =17, fontsize_col = 15, margins=c(7,3),color=hmcols,legend_labels =T)

```


```{r}

data=data_beforescale

# Select data for a single sample from data containing read per TE cluster of all sample
v_194MR=data.frame(data$v_194MR)
# Add TE cluster names as rownames
rownames(v_194MR)=rownames(data)
# Remove rows with TE families without read coverage
v_194MR=v_194MR%>% filter(data.v_194MR>0)
v_194MR$level=NA
# Adding a factor : high:  if the number of reads for the cluster is higher than mean number of reads of all TE cluster and low: if not 
v_194MR[v_194MR$data.v_194MR>=mean(v_194MR$data.v_194MR),]$level<-"high"
v_194MR[v_194MR$data.v_194MR<mean(v_194MR$data.v_194MR),]$level<-"low"

# plot a dotplot of RPKM of mobilome reads for each TE family 
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("194MR_dotplot.png",res=700,width=17000,height=6000)
ggplot(v_194MR,aes(x=rownames(v_194MR),y=data.v_194MR,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5))  

# Select data for a single sample from data containing read per TE cluster of all sample
v_194MR=data.frame(data$v_194MR)
# Add TE cluster names as rownames
rownames(v_194MR)=rownames(data)
# plot a dotplot of RPKM of mobilome reads for each TE family 
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("194MR_dotplot.png",res=700,width=17000,height=6000)
ggplot(v_194MR,aes(x=reorder(rownames(v_194MR),-data.v_194MR),y=data.v_194MR))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5))  

v_216MR=data.frame(data$v_216MR)
rownames(v_216MR)=rownames(data)
v_216MR=v_216MR%>% filter(data.v_216MR>0)
v_216MR$level=NA
v_216MR[v_216MR$data.v_216MR>=mean(v_216MR$data.v_216MR),]$level<-"high"
v_216MR[v_216MR$data.v_216MR<mean(v_216MR$data.v_216MR),]$level<-"low"
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("216MR_dotplot.png",res=700,width=17000,height=6000)
ggplot(v_216MR,aes(x=rownames(v_216MR),y=data.v_216MR,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 

v_216MR=data.frame(data$v_216MR)
rownames(v_216MR)=rownames(data)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("216MR_dotplot.png",res=700,width=17000,height=6000)
ggplot(v_216MR,aes(x=reorder(rownames(v_216MR),-data.v_216MR),y=data.v_216MR))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5))  


BA20=data.frame(data$BA20)
rownames(BA20)=rownames(data)
BA20=BA20%>% filter(data.BA20>0)
BA20$level=NA
BA20[BA20$data.BA20>=mean(BA20$data.BA20),]$level<-"high"
BA20[BA20$data.BA20<mean(BA20$data.BA20),]$level<-"low"
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("BA20_dotplot.png",res=700,width=17000,height=6000)
ggplot(BA20,aes(x=rownames(BA20),y=data.BA20,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 
BA20=data.frame(data$BA20)
rownames(BA20)=rownames(data)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("BA20_dotplot.png",res=700,width=17000,height=6000)
ggplot(BA20,aes(x=reorder(rownames(BA20),-data.BA20),y=data.BA20))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5))  


V25=data.frame(data$V25)
rownames(V25)=rownames(data)
V25=V25%>% filter(data.V25>0)
V25$level=NA
V25[V25$data.V25>=mean(V25$data.V25),]$level<-"high"
V25[V25$data.V25<mean(V25$data.V25),]$level<-"low"
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("V25_dotplot.png",res=700,width=17000,height=6000)
ggplot(V25,aes(x=rownames(V25),y=data.V25,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 

V25=data.frame(data$V25)
rownames(V25)=rownames(data)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("V25_dotplot.png",res=700,width=17000,height=6000)
ggplot(V25,aes(x=reorder(rownames(V25),-data.V25),y=data.V25))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5))  


CC9=data.frame(data$CC9)
rownames(CC9)=rownames(data)
CC9=CC9%>% filter(data.CC9>0)
CC9$level=NA
CC9[CC9$data.CC9>=mean(CC9$data.CC9),]$level<-"high"
CC9[CC9$data.CC9<mean(CC9$data.CC9),]$level<-"low"
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("CC9_dotplot.png",res=700,width=17000,height=6000)
ggplot(CC9,aes(x=rownames(CC9),y=data.CC9,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 
CC9=data.frame(data$CC9)
rownames(CC9)=rownames(data)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("CC9_dotplot.png",res=700,width=17000,height=6000)
ggplot(CC9,aes(x=reorder(rownames(CC9),-data.CC9),y=data.CC9))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5))  


BD18=data.frame(data$BD18)
rownames(BD18)=rownames(data)
BD18=BD18%>% filter(data.BD18>0)
BD18$level=NA
BD18[BD18$data.BD18>=mean(BD18$data.BD18),]$level<-"high"
BD18[BD18$data.BD18<mean(BD18$data.BD18),]$level<-"low"
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("BD18_dotplot.png",res=700,width=17000,height=6000)
ggplot(BD18,aes(x=rownames(BD18),y=data.BD18,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 

BD18=data.frame(data$BD18)
rownames(BD18)=rownames(data)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("BD18_dotplot.png",res=700,width=17000,height=6000)
ggplot(BD18,aes(x=reorder(rownames(BD18),-data.BD18),y=data.BD18))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 

BV15=data.frame(data$BV15)
rownames(BV15)=rownames(data)
BV15=BV15%>% filter(data.BV15>0)
BV15$level=NA
BV15[BV15$data.BV15>=mean(BV15$data.BV15),]$level<-"high"
BV15[BV15$data.BV15<mean(BV15$data.BV15),]$level<-"low"
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("BV15_dotplot.png",res=700,width=17000,height=6000)
ggplot(BV15,aes(x=rownames(BV15),y=data.BV15,col=level))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) 

BV15=data.frame(data$BV15)
rownames(BV15)=rownames(data)
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("BV15_dotplot.png",res=700,width=17000,height=6000)
ggplot(BV15,aes(x=reorder(rownames(BV15),-data.BV15),y=data.BV15))+geom_point(size=0.8) + scale_colour_manual(values = c("red","grey")) + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1.5)) +geom_hline(yintercept=mean(BV15$data.BV15))

library("in2extRemes")
fitGEV <- fevd(data.BV15, data = BV15)
fitGEV
plot(fitGEV)


# Select a threshold for daily data from above.
threshrange.plot(BV15$data.BV15, c(0.1, 2), type = "PP")

# Fit the PP model to the daily data from above.
fitPP <- fevd(data.BV15, data = BV15, threshold = 0.395,
        type = "PP", units = "inches")
fitPP
plot(fitPP)

atdf(BV15$data.BV15, 0.395)

extremalindex(BV15$data.BV15, 0.5)

dcFort <- decluster(BV15$data.BV15, 0.395, r = 9)
plot(dcFort)

```

```{r}
library(PCAtools)
library(DESeq2)
library("EnhancedVolcano")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")

# Format the numerical matrix of mobilome reads for all sample to be used in DESEQ2 analysis
data=rd_data_lr
data=data[,c(1:25)]
data=data[,-c(19)]
#data=data[,-c(10,12)]

data=data.frame(column_to_rownames(data, var = "name"))
data=as.matrix(data)
colnames(data)<-c("194MR","216MR","BA20","V25","CC9","BD18","BV15","185M","354R","CBRR","525M","525R","GE1","BD18_2","BV15_2","CC9_2","V25_2","216M","CC9_b","H8007","BA20_b","BD18_b","V25_b")
data=subset( data, select = -c(BD18_2,BV15_2,CC9_2,V25_2,CBRR))

#colnames(data)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","v185M_barcode01","v525M_barcode04","GE1_barcode06","BD18_barcode07","BV15_barcode08","CC9_barcode09","V25_barcode10","216M_barcode11")
#colnames(data)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","barcode01","barcode02","barcode04","barcode05","barcode06","barcode07","barcode08","barcode09","barcode10","barcode11","barcode12")
# Format conditions data for DESEQ2 analysis
#condition=c("verzy","massane")
condition=c("Bud","Leaf")
#condition=c("MR","M","R","massane")
#coldata=data.frame(c("verzy","verzy","massane","massane","massane","massane","massane","verzy","verzy","verzy","verzy","massane","massane","massane","massane","massane","verzy"))
#coldata=data.frame(c("b","b","f","f","f","f","f","b","b","b","b","b","f","f","f","f","f","f"))
#coldata=data.frame(c("Bud","Bud","Leaf","Leaf","Leaf","Leaf","Leaf","Bud","Bud","Bud","Bud","Bud","Leaf","Leaf"))
coldata=data.frame(c("Bud","Bud","Leaf","Leaf","Leaf","Leaf","Leaf","Bud","Bud","Bud","Bud","Leaf","Leaf","Bud","Bud","Bud","Bud","Bud"))
#coldata=data.frame(c("MR","MR","massane","massane","massane","massane","massane","M","R","M","R","massane","massane","massane","massane","massane","M"))
colnames(coldata)="condition"
#rownames(coldata)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","v185M_barcode01","v525M_barcode04","GE1_barcode06","BD18_barcode07","BV15_barcode08","CC9_barcode09","V25_barcode10","216M_barcode11")
#rownames(coldata)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","barcode01","barcode02","barcode03","barcode04","barcode05","barcode06","barcode07","barcode08","barcode09","barcode10","barcode11")

#rownames(coldata)<-c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","185M","354R","CBRR","525M","525R","GE1","BD18_2","BV15_2","CC9_2","V25_2","216M")
#rownames(coldata)<-c("194MR","216MR","BA20","V25","CC9","BD18","BV15","185M","354R","CBRR","525M","525R","GE1","216M")
rownames(coldata)<-c("194MR","216MR","BA20","V25","CC9","BD18","BV15","185M","354R","525M","525R","GE1","216M","CC9_b","H8007","BA20_b","BD18_b","V25_b")
# Combine condition data and read matrix
data=data+1
dds=DESeqDataSetFromMatrix(countData=data,colData=coldata,design = ~ condition)
# Call differential expression analysis 
dds <- DESeq(dds)
# Retriving results with pvalue
res <- results(dds,alpha=0.05,pAdjustMethod="none")

# Perform PCA analysis
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
pcaData <- plotPCA(vsd, intgroup=c("condition"),returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

# Plot PCA 
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("PCA_lr1.png",res=600,width=4500,height = 3500)
#ggplot(pcaData, aes(PC1, PC2, color=condition))  + geom_point(size=16) + xlab(paste0("PC1: ",percentVar[1],"% variance")) + ylab(paste0("PC2: ",percentVar[2],"% variance")) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12))  + geom_text(aes(label = name), nudge_x = 0,nudge_y=0, color="white",size=2.5)  + scale_color_manual(values=c("seagreen", "red3","green3","red3")) + theme_classic()

ggplot(pcaData, aes(PC1, PC2,col=condition,label = name))  + geom_point(size=14) + xlab(paste0("PC1: ",percentVar[1],"% variance"))  + ylab(paste0("PC2: ",percentVar[2],"% variance")) + theme(axis.text=element_text(size=12.5),axis.title=element_text(size=4))  + scale_color_manual(values=c("yellow4", "coral4")) + theme_classic() + geom_label(mapping = aes(label = name),max.overlaps=30,size=2) + theme(axis.line=element_line(size=0.6))  #+
  #geom_text_repel(aes(label = name),color = "black", nudge_y = 0,  nudge_x = 0, segment.size = 0, force = 1, max.overlaps = Inf)  
 # + geom_label(mapping = aes(label = name),max.overlaps=30,size=2) 
# + geom_label(mapping = aes(label = name),max.overlaps=30,size=2) 
```


```{r}

library(PCAtools)
library(DESeq2)
library("EnhancedVolcano")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")

# Format the numerical matrix of mobilome reads for all sample to be used in DESEQ2 analysis
data=rd_data_lr
data=data[,c(1:15)]
data=data.frame(column_to_rownames(data, var = "name"))
data=as.matrix(data)

# Format conditions data for DESEQ2 analysis
condition=c("verzy","massane")
coldata=data.frame(c("verzy","verzy","massane","massane","massane","massane","massane","verzy","verzy","verzy","massane","massane","massane","massane"))
colnames(coldata)="condition"
rownames(coldata)=c("v_194MR","v_216MR","BA20","V25","CC9","BD18","BV15","barcode01","barcode03","barcode04","barcode06","barcode07","barcode08","barcode09")

# Combine condition data and read matrix
dds=DESeqDataSetFromMatrix(countData=data,colData=coldata,design = ~ condition)
# Call differential expression analysis 
dds <- DESeq(dds)
# Retriving results with pvalue
res <- results(dds,alpha=0.05,pAdjustMethod="none")

# Perform PCA analysis
vsd <- varianceStabilizingTransformation(dds, blind=FALSE)
pcaData <- plotPCA(vsd, intgroup=c("condition"),returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

# Plot PCA 
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("PCA_lr1.png",res=600,width=5000,height = 4000)
ggplot(pcaData, aes(PC1, PC2, color=condition,label = name))  + geom_point(size=16) + xlab(paste0("PC1: ",percentVar[1],"% variance")) + ylab(paste0("PC2: ",percentVar[2],"% variance")) + theme(axis.text=element_text(size=12),axis.title=element_text(size=12))  + geom_text(aes(label = name), nudge_x = 0,nudge_y=0, color="red",size=2.5)  + scale_color_manual(values=c("navy", "gold1")) + theme_classic()

```



```{r}
library("readxl")
setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data/")
annot_te=read_excel("/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_TE_stats_classif.xlsx")
annot_te=annot_te[,c(1,6,7,12:15)]

cluster_te=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
data=left_join(annot_te,cluster_te,by=c("Seq_name Cur4"="V1"))

data_mite=data[data$order=="MITE",]
data_mite=unique(data_mite[,c(1,8)])
write.table(data_mite,"MITE_FsylCur4_denovoLibTEs_cluster.fa.clustr",row.names = F,col.names = F,quote=F,sep="\t")
data$proportion_compl=data$fullLgthCopies/data$copies


library("readxl")
setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data/")
annot_te=read_excel("/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_TE_stats_classif.xlsx")
annot_te=annot_te[,c(1,6,7,12:15)]

cluster_te=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
data=left_join(annot_te,cluster_te,by=c("Seq_name Cur4"="V1"))

data_tir=data[data$order=="TIR",]
data_tir=unique(data_tir[,c(1,8)])
write.table(data_tir,"TIR_FsylCur4_denovoLibTEs_cluster.fa.clustr",row.names = F,col.names = F,quote=F,sep="\t")
data$proportion_compl=data$fullLgthCopies/data$copies
data1=data[data$copies>=10,]


```


```{r}
library(tidyverse)
library(janitor)
library(rstatix)
library(ggpubr)
library(scales)

data=rd_data_lr
data$name=str_replace_all(data$name,"helitron","Helitron")
data$name=str_replace_all(data$name,"MITE|MITE_Cluster_1457","MITE_Cluster_1457")
data$name=str_replace_all(data$name,"MITE|MITE_Cluster_1585","MITE_Cluster_1585")
data$name=str_replace_all(data$name,"\\|NA","")
data$fam=str_split_fixed(data$name,"_",2)[,1]

size=unique(data[,c(9,10)])
data=data[,c(10,2:8)]

data=data %>% group_by(fam) %>% summarise(across(everything(), sum))
size=size %>% group_by(fam) %>% summarise(mean = median(size))

data$v_194MR=data$v_194MR/0.215872
data$v_216MR=data$v_216MR/0.190463
data$BA20=data$BA20/0.138765
data$V25=data$V25/0.611088
data$CC9=data$CC9/0.793042
data$BD18=data$BD18/0.695400
data$BV15=data$BV15/0.022979

data=left_join(data,size,by=c("fam"="fam"))
data$v_194MR=data$v_194MR/data$mean
data$v_216MR=data$v_216MR/data$mean
data$BA20=data$BA20/data$mean
data$CC9=data$CC9/data$mean
data$V25=data$V25/data$mean
data$BD18=data$BD18/data$mean
data$BV15=data$BV15/data$mean

data=data[,c(1:8)]
library(reshape2)
data1 <- na.omit(melt(data))

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("barplot_te.png",width=9000,height = 6000,res=400)
ggplot(data1, aes(x=variable,y=value, fill=fam))  + geom_col(position = "dodge") 

#setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
#png("boxplot_te.png",width=9000,height = 6000,res=400)
#ggplot(data1, aes(x=fam,y=value,fill=fam)) + geom_boxplot() 


library(RColorBrewer)


data2 <- data1 %>% 
  group_by(variable) %>% 
  mutate(perc = (value / sum(value))*100) 

data2$fam[(data2$fam=="Helitron" | data2$fam=="LINE|SINE" | data2$fam=="SINE" | data2$fam=="LARD" | data2$fam=="TRIM")]= "Others"
data2 <- data2 %>% 
  group_by(variable,fam) %>% 
  mutate(perc = sum(perc)) 
data2$fam=as.factor(data2$fam)
data2$perc=round(data2$perc,digits=1)

data2=data.frame(unique(data2[,c(1,2,4)]))


setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("pieplot_te.png",width=9000,height = 6000,res=400)
#+ geom_text(aes(label = fam),position = position_stack(vjust = 0.5))
ggplot(data2, aes(x = "", y = perc, fill = as.factor(fam))) + geom_bar(stat = "identity",color = "black",width=1) + geom_text(aes(x=1.25,label=paste0(round(perc, 1), "%")),position = position_stack(vjust = 0.5)) +facet_grid(.~variable) + coord_polar(theta = "y",start=0) + scale_fill_brewer(palette="Set2") + theme(legend.position = "top",legend.justification='left',legend.direction='horizontal',legend.box.margin = margin(5, l = 5, unit = "mm")) + theme_linedraw() + theme(axis.text = element_blank(), axis.ticks = element_blank(), panel.grid  = element_blank()) 


```


```{r}
library(karyoploteR) 
genome=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/30scaff.sizes.genome")
custom.genome <- toGRanges(genome)
gene=read.table(file="/home/lgdp/Desktop/gene_TIP_upst_downst/Fagus_Sylvatica_gene_annot_fn2.bed")
gene=toGRanges(gene[,(1:4)])

bed=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster59_e-50_d50.bed")
#ind=read.table(file="SPRITE1_mobilome_peaks_nb_ind_pos.bed")

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("karyoplot_SPRITE1.png",width=9000,height = 6000,res=400)

kp <- plotKaryotype(genome=custom.genome,cex=0.85, cytobands =gene,plot.type = 2)
kpAddBaseNumbers(kp, tick.dist = 5000000, tick.len = 40, tick.col="gray3", cex=0.6,
                 minor.tick.dist = 500000, minor.tick.len = 25, minor.tick.col = "gray3")
kpArrows(kp, chr=bed$V1, x0=bed$V2,x1=bed$V3,y0=0.5,y1=-0.05,length=0.06,data.panel = 2,col="blue3")


library(karyoploteR) 
genome=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/30scaff.sizes.genome")
custom.genome <- toGRanges(genome)
bed=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster52_e-50_d50.bed")
#ind=read.table(file="SPRITE1_mobilome_peaks_nb_ind_pos.bed")

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("karyoplot_Cluster52.png",width=9000,height = 6000,res=400)
kp <- plotKaryotype(genome=custom.genome,plot.type = 2,cex=0.85,cytobands =gene)
kpAddBaseNumbers(kp, tick.dist = 5000000, tick.len = 40, tick.col="gray3", cex=0.6,
                 minor.tick.dist = 500000, minor.tick.len = 25, minor.tick.col = "blue")

kpArrows(kp, chr=bed$V1, x0=bed$V2,x1=bed$V3,y0=0.5,y1=-0.05,length=0.06,data.panel = 2,col="blue3")




library(karyoploteR) 
genome=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/30scaff.sizes.genome")
custom.genome <- toGRanges(genome)
bed=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster33_e-50_d50.bed")
#ind=read.table(file="SPRITE1_mobilome_peaks_nb_ind_pos.bed")

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("karyoplot_Cluster33.png",width=9000,height = 6000,res=400)
kp <- plotKaryotype(genome=custom.genome,plot.type = 2,cex=0.85,cytobands =gene)
kpAddBaseNumbers(kp, tick.dist = 5000000, tick.len = 40, tick.col="gray3", cex=0.6,
                 minor.tick.dist = 500000, minor.tick.len = 25, minor.tick.col = "blue")

kpArrows(kp, chr=bed$V1, x0=bed$V2,x1=bed$V3,y0=0.5,y1=-0.05,length=0.06,data.panel = 2,col="blue3")




library(karyoploteR) 
genome=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/30scaff.sizes.genome")
custom.genome <- toGRanges(genome)
bed=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster1249_e-50_d50.bed")
#ind=read.table(file="SPRITE1_mobilome_peaks_nb_ind_pos.bed")

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("karyoplot_1249.png",width=9000,height = 6000,res=400)
kp <- plotKaryotype(genome=custom.genome,plot.type = 2,cex=0.85,cytobands =gene)
kpAddBaseNumbers(kp, tick.dist = 5000000, tick.len = 40, tick.col="gray3", cex=0.6,
                 minor.tick.dist = 500000, minor.tick.len = 25, minor.tick.col = "blue")

kpArrows(kp, chr=bed$V1, x0=bed$V2,x1=bed$V3,y0=0.5,y1=-0.05,length=0.06,data.panel = 2,col="blue3")




library(karyoploteR) 
genome=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/30scaff.sizes.genome")
custom.genome <- toGRanges(genome)
bed=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster1564_e-50_d50.bed")
#ind=read.table(file="SPRITE1_mobilome_peaks_nb_ind_pos.bed")

setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("karyoplot_Cluster1564.png",width=9000,height = 6000,res=400)
kp <- plotKaryotype(genome=custom.genome,plot.type = 2,cex=0.85,cytobands =gene)
kpAddBaseNumbers(kp, tick.dist = 5000000, tick.len = 40, tick.col="gray3", cex=0.6,
                 minor.tick.dist = 500000, minor.tick.len = 25, minor.tick.col = "blue")

kpArrows(kp, chr=bed$V1, x0=bed$V2,x1=bed$V3,y0=0.5,y1=-0.05,length=0.06,data.panel = 2,col="blue3")



#kpDataBackground(kp, data.panel = 1, col="white")
#kpPlotRegions(kp, data=bed,col="palevioletred3", data.panel = 2)

#kpAxis(kp, ymin = 0, ymax=6, cex=0.5)

#point.colors2 <- colByValue(ind$V4, colors = c("green","red3"), min = 0, max=6)
#kpPoints(kp,data=toGRanges(ind), y=ind$V4,col=point.colors2,ymin=0, ymax=6)

```
```{r}

data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster33_e-50_d50.bed")
data$size=data$V3-data$V2
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust33_size.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=size))+geom_density(fill="royalblue3")+scale_x_continuous(limits=c(0,500),breaks = seq(0,500,100))+ theme_classic() +theme(axis.text.y = element_text(size=10),axis.text.x = element_text(size=10))


data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster52_e-50_d50.bed")
data$size=data$V3-data$V2
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust52_size.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=size))+geom_density(fill="royalblue3")+scale_x_continuous(limits=c(0,3800),breaks = seq(0,3800,400))+ theme_linedraw()

data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster59_e-50_d50.bed")
data$size=data$V3-data$V2
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust59_size.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=size))+geom_density(fill="royalblue3")+scale_x_continuous(limits=c(0,900),breaks = seq(0,900,100))+ theme_classic() +theme(axis.text.y = element_text(size=20),axis.text.x = element_text(size=20))

data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster1249_e-50_d50.bed")
data$size=data$V3-data$V2
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust1249_size.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=size))+geom_density(fill="royalblue3")+scale_x_continuous(limits=c(0,600),breaks = seq(0,600,100))+ theme_linedraw()


data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster1564_e-50_d50.bed")
data$size=data$V3-data$V2
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust1564_size.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=size))+geom_density(fill="royalblue3")+scale_x_continuous(limits=c(0,400),breaks = seq(0,400,100))+ theme_linedraw()

```

```{r}

setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data/")
data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster33_e-50_d50_gene_dist.bed")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust33_gene_dist.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=V4))+geom_density(fill="palevioletred2")+scale_x_continuous(limits=c(-190000,120000),breaks = seq(-190000,120000,50000))+ theme_classic()

setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data")
data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster52_e-50_d50_gene_dist.bed")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust52_gene_dist.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=V4))+geom_density(fill="palevioletred2")+scale_x_continuous(limits=c(-240000,120000),breaks = seq(-240000,120000,50000))+ theme_linedraw()

setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data")
data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster59_e-50_d50_gene_dist.bed")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust59_gene_dist.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=V4))+geom_density(fill="palevioletred2")+scale_x_continuous(limits=c(-80000,45000),breaks = seq(-80000,45000,10000))+ theme_classic() +theme(axis.text.y = element_text(size=20),axis.text.x = element_text(size=20))

setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data")
data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster1249_e-50_d50_gene_dist.bed")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust1249_gene_dist.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=V4))+geom_density(fill="palevioletred2")+scale_x_continuous(limits=c(-140000,170000),breaks = seq(-140000,1700000,50000))+ theme_linedraw()

setwd("/home/lgdp/Desktop/FsylCur4_last_annotation_data")
data=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/Cluster1564_e-50_d50_gene_dist.bed")
setwd("/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome")
png("Clust1564_gene_dist.png",width=9000,height = 6000,res=400)
ggplot(data,aes(x=V4))+geom_density(fill="palevioletred2")+scale_x_continuous(limits=c(-110000,320000),breaks = seq(-110000,320000,50000))+ theme_linedraw()

```



