

```{r}
library("ggplot2")

setwd("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP")

data=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/out.hwe",header=T)
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/struct_pop.png",res=600,width=6000,height = 3000)
ggplot(data,aes(x=P_HWE))+geom_histogram(col="black")+theme(axis.line=element_line(size=0.75))+theme_bw()
ggplot(data,aes(x=P_HET_DEFICIT))+geom_histogram(col="black") +theme(axis.line=element_line(size=0.75))+theme_bw()
ggplot(data,aes(x=P_HET_EXCESS))+geom_histogram(col="black") +theme(axis.line=element_line(size=0.75))+theme_bw()

data=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/Beech_147_complet_23062022.recode.Freq.txt",header=T)
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/SNP_freq_histo.png",res=600,width=4000,height = 5000)
ggplot(data,aes(x=X44))+geom_density(col="black",fill="azure4")+theme(axis.line=element_line(size=0.75))+theme_bw()

data=read.table(file="/home/lgdp/Downloads/Beach_Massane_coverage.tab",header=T)
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/coverage_table_147_ind.png",res=600,width=4000,height = 5000)
ggplot(data,aes(y=cov))+geom_boxplot()+theme(axis.line=element_line(size=0.75))+theme_bw()


data=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/Beech_147_complet_23062022.recode.NbSNP_perInd.txt")
mean(data$V2)
data$V3="ind"
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/SNP_per_Ind_violin.png",res=600,width=4000,height = 5000)
ggplot(data,aes(x=V3,y=V2))+geom_violin(fill="azure4")+theme(axis.line=element_line(size=0.75))+theme_bw()

png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/SNP_per_Ind_barplot.png",res=600,width=8000,height = 4000)
ggplot(data,aes(x=V1,y=V2))+geom_bar(stat="identity",col="black",fill="azure")+theme(axis.line=element_line(size=0.75))+theme_bw()+  theme(axis.text.x = element_text(angle = 90,size=7.5))

```



```{r}
library("ggplot2")
library("dplyr")
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data=read.table(file="all_LTR_ins_TE_ind.bed")

data1=data %>% group_by(V4,V5) %>% summarise(count=n())

data2 <- data1 %>% group_by(V4) %>%  mutate(mean = mean(count))
data2=data2[data2$mean>=20,]
data3=data1[data1$V4 %in% data2$V4,]

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/boxplot_TEs_per_ind_per_family.png",res=800,width=18500,height = 7000)
ggplot(data3,aes(x=V4,y=count)) + geom_boxplot(outlier.size=0.01) +theme_bw() + theme(axis.text.x = element_text(angle = 45, hjust = 1,size=5),axis.text.y = element_text( hjust = 1,size=10),axis.title = element_text(size = 18),axis.line=element_line(size=0.75)) + stat_summary(fun.y=mean, geom="point", shape=8, size=1.5, color="red2", fill="red2") + labs(x= "TE family",y= "Number of insertions")  + scale_y_continuous(breaks = seq(0, 4000, by = 500)) 



data=data3[,c(2,3)]
data=data%>% group_by(V5) %>% summarise(Frequency = sum(count))

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_ins_LTR_ind_barplot.png",res=600,width=8000,height = 4000)
ggplot(data,aes(x=V5,y=Frequency))+geom_bar(stat="identity",col="black",fill="royalblue4")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5))

data$V3="ind"
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_ins_LTR_ind_violin.png",res=600,width=6000,height = 5000)
ggplot(data,aes(x=V3,y=Frequency))+geom_violin(col="black",fill="royalblue4")+theme(axis.line=element_line(size=0.75))+theme_bw()

data=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_family_windows_matrics_count_TEfam.txt",sep=" ")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_LTR_fam_bar.png",res=600,width=15000,height = 8000)
data$V3="ind"
ggplot(data,aes(x=V1,y=V2))+geom_bar(col="black",fill="royalblue4",stat="identity")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=3.5))


data=read.table(file="allTIPs_pop.bed")
data$freq=data$V5/147
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/TIPs_LTR_freq_density.png",res=600,width=10000,height = 5000)
ggplot(data,aes(x=freq))+geom_histogram(col="black",fill="royalblue4") +theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=1))+theme_bw() + scale_x_continuous(breaks = seq(0,1,0.1))


##################################################################
library("ggplot2")
library("dplyr")
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data0=read.table(file="all_LTRs_ins_TE_ind2.csv2.bed",header=T)
ind=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2_ind.txt")
colnames(data0)=ind
data=data0

data0=data0[,c(1,2)]
data0= data0 %>% group_by(TE) %>%summarize(count = n())
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_LTR_fam_bar.png",res=600,width=15000,height = 8000)
ggplot(data0,aes(x=reorder(TE,-count),y=count))+geom_bar(col="black",fill="royalblue4",stat="identity")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 1, hjust=1,size=3))

data0$TE=str_replace_all(data0$TE, "LTR", "retrofag")  
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_LTR_fam_bar_1.png",width=7000,height = 5000,res=500)
ggplot(data0[data0$count>=1367,],aes(x=reorder(TE,-count),y=count))+ geom_point(size=3, shape=23,fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 1, hjust=1,size=13))

id=paste0(data$TE,"_",data$ins)
rownames(data)=id
data=data[,-c(1,2)]

data1=data.frame(colSums(data)) 
data1$ind=row.names(data1)
colnames(data1)=c("nb","ind")

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_ins_LTR_ind_barplot.png",res=600,width=8000,height = 4000)
ggplot(data1,aes(x=ind,y=nb))+geom_bar(stat="identity",col="black",fill="royalblue4")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5))

data1$V3="ind"
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_ins_LTR_ind_violin.png",res=600,width=6000,height = 5000)
ggplot(data1,aes(x=V3,y=nb))+geom_violin(col="black",fill="royalblue4")+theme(axis.line=element_line(size=0.75))+theme_bw()

data1=data.frame(rowSums(data)) 
data1$ind=row.names(data1)
colnames(data1)=c("freq","win")
data1$freq=data1$freq/147
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_ins_LTR_freq_density.png",res=600,width=8000,height = 4000)
ggplot(data1,aes(x=freq))+geom_density(col="black",fill="royalblue4")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5))

data=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_win_genes.bed")
data$win=paste0(data$V5,"_",data$V6,"_",data$V7,"_",data$V8)
data=data[,c(10,9)]
data$V9=data$V9/100

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_win_gene_dist.png",res=600,width=15000,height = 8000)
ggplot(data,aes(x=V9))+geom_density(col="black",fill="royalblue4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=3.5)) + scale_x_continuous(breaks = seq(-2000,3000,100))

data=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/genes_LTR_tips_count.txt")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/genes_LTR_tips_count_barplot.png",res=600,width=9000,height = 7000)
ggplot(data[data$V2>=115,],aes(x=reorder(V1,V2),y=V2))+geom_bar(stat="identity",col="black",fill="royalblue4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/genes_MITE_tips_count.txt")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/genes_MITE_tips_count_barplot.png",res=600,width=9000,height = 7000)
ggplot(data1[data1$V2>=67,],aes(x=reorder(V1,V2),y=V2))+geom_bar(stat="identity",col="black",fill="purple4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) +coord_flip()

data2=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/Fagus_sylvatica_v3.1.annot.simple_nb_SNP.bed")
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/genes_SNP_count_barplot.png",res=600,width=9000,height = 7000)
ggplot(data2[data2$V5>=1352,],aes(x=reorder(V4,V5),y=V5))+geom_bar(stat="identity",col="black",fill="darkorange3")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) +coord_flip()

gene=data.frame(unique(c(data$V1,data1$V1)))
colnames(gene)="V1"
data=left_join(gene,data,by=c("V1"="V1"))
data=left_join(data,data1,by=c("V1"="V1"))

library(reshape2) 
colnames(data)=c("gene","LTR","MITE")
data1=melt(data,id.vars=c("gene"),measure.vars=c("LTR","MITE"))







setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
data0=read.table(file="all_MITEs_ins_TE_ind2.csv2.bed",header=T)
ind=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2_ind.txt")
colnames(data0)=ind
data=data0

data0=data0[,c(1,2)]
data0= data0 %>% group_by(TE) %>%summarize(count = n())

mite=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
data0=left_join(data0,mite,by=c("TE"="V1"))

for (i in 1:nrow(data0)){
  if (is.na(data0$V2[i])) {
    data0$V2[i]<-data0$TE[i]
  }
}
data0=data0[,-c(1)]

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_TIPs_MITE_fam_bar.png",res=600,width=15000,height = 8000)
ggplot(data0,aes(x=reorder(V2,-count),y=count))+geom_bar(col="black",fill="purple4",stat="identity")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45,size=3.5,vjust = 1, hjust=1))


png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_MITE_fam_bar_1.png",width=7000,height = 5000,res=500)
ggplot(data0[data0$count>=1442,],aes(x=reorder(V2,-count),y=count))+ geom_point(size=3, shape=23,fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 1, hjust=1,size=13))


id=paste0(data$TE,"_",data$ins)
rownames(data)=id
data=data[,-c(1,2)]
data1=data.frame(colSums(data)) 
data1$ind=row.names(data1)
colnames(data1)=c("nb","ind")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_ins_MITE_ind_barplot.png",res=600,width=8000,height = 4000)
ggplot(data1,aes(x=ind,y=nb))+geom_bar(stat="identity",col="black",fill="purple4")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5))

data1$V3="ind"
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_ins_MITE_ind_violin.png",res=600,width=6000,height = 5000)
ggplot(data1,aes(x=V3,y=nb))+geom_violin(col="black",fill="purple4")+theme(axis.line=element_line(size=0.75))+theme_bw()

data1=data.frame(rowSums(data)) 
data1$ind=row.names(data1)
colnames(data1)=c("freq","win")
data1$freq=data1$freq/147
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_ins_MITE_freq_density.png",res=600,width=8000,height = 4000)
ggplot(data1,aes(x=freq))+geom_density(col="black",fill="purple4")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5))

data=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv_win_genes.bed")
data$win=paste0(data$V5,"_",data$V6,"_",data$V7,"_",data$V8)
data=data[,c(10,9)]
data$V9=data$V9/100
library(ggplot2)
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv_win_gene_dist.png",res=600,width=15000,height = 8000)
ggplot(data,aes(x=V9))+geom_density(col="black",fill="purple4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=3.5))  + scale_x_continuous(breaks = seq(-3100,3700,100))




```


```{r}
setwd("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data0=read.table(file="all_LTRs_ins_TE_ind2.csv2.bed",header=T)
ind=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2_ind.txt")
colnames(data0)=ind
data0=data0[,c(1,2)]
data0= data0 %>% group_by(TE) %>%summarize(count = n())
 
png("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_LTR_fam_bar_1.png",width=7000,height = 5000,res=500)
ggplot(data0[data0$count>=1367,],aes(x=reorder(TE,-count),y=count))+ geom_point(size=3, shape=23,fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 1, hjust=1,size=13))

setwd("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data=read.table(file="all_LTR_ins_TE_ind_removed_filt.bed")
data1=data %>% group_by(V4,V5) %>% summarise(count=n())
data2 <- data1 %>% group_by(V4) %>%  mutate(mean = mean(count))
data2=unique(data2[,c(1,4)])
png("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_LTR_nb_copy.png",res=600,width=15000,height = 8000)
ggplot(data2[data2$mean>=279,],aes(x=reorder(V4,-mean),y=mean)) + geom_point(col="black",size=6) +theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11)) 
data=left_join(data0,data2,by=c("TE"="V4"))
png("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_LTR_nb_copy_tips.png",res=600,width=15000,height = 8000)
ggplot(data,aes(y=count,x=mean)) + geom_point(col="black",size=6) +theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11)) + geom_smooth()




setwd("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data0=read.table(file="all_MITEs_ins_TE_ind2.csv2.bed",header=F)
ind=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2_ind.txt")
colnames(data0)=ind
data0=data0[,c(1,2)]

annot=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
data0=left_join(data0,annot,by=c("TE"="V1"))

data0[is.na(data0$V2),]$V2<-data0[is.na(data0$V2),]$TE
data0=data0[,c(2,3)]
data0= data0 %>% group_by(V2) %>%summarize(count = n())
 
png("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_MITE_fam_bar_1.png",width=7000,height = 5000,res=500)
ggplot(data0[data0$count>=1367,],aes(x=reorder(V2,-count),y=count))+ geom_point(size=3, shape=23,fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 1, hjust=1,size=13))

setwd("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data=read.table(file="all_MITE_ins_TE_ind_removed_filt.bed")
data=left_join(data,annot,by=c("V4"="V1"))
data[is.na(data$V2.y),]$V2.y<-data[is.na(data$V2.y),]$V4

data=data[,c(1:3,5,6)]

data1=data %>% group_by(V5,V2.y) %>% summarise(count=n())
data2 <- data1 %>% group_by(V2.y) %>%  mutate(mean = mean(count))
data2=unique(data2[,c(2,4)])
png("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_MITE_nb_copy.png",res=600,width=15000,height = 8000)
ggplot(data2[data2$mean>=279,],aes(x=reorder(V2.y,-mean),y=mean)) + geom_point(col="black",size=6) +theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11)) 
data=left_join(data2,data0,by=c("V2.y"="V2"))
png("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_MITE_nb_copy_tips.png",res=600,width=15000,height = 8000)
ggplot(data,aes(y=count,x=mean)) + geom_point(col="black",size=6) +theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11)) + geom_smooth()  +geom_text(aes(label=ifelse(PTS>24,as.character(Name),'')),hjust=0,vjust=0)




library("ggplot2")
library("dplyr")
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
ind=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2_ind.txt")
data0=read.table(file="all_LTRs_ins_TE_ind2.csv2.bed",header=F)
colnames(data0)=ind

sub=data0[,c(1,2)]
ltr_sub2 = sub %>% group_by(ins) %>% summarize(Count = n())
ltr_sub2$te="ltr"
data0=data0[,-c(1,2)]
data0=data.frame(colSums(data0)) 
data0$ind=rownames(data0)
data0$ltr="ltr"
colnames(data0)=c("nb","ind","te")


ind=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2_ind.txt")
data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv2.bed",header=F)
colnames(data1)=ind
sub=data1[,c(1,2)]
colnames(sub)=c("TE","ins")
mite_sub2 = sub %>% group_by(ins) %>% summarize(Count = n())
mite_sub2$te="mite"
sub2=rbind(ltr_sub2,mite_sub2)
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_MITE_LTR_TIPs_boxplot.png",res=600,width=15000,height = 8000)
ggplot(sub2,aes(y=log(Count),fill=te)) + geom_boxplot(col="black",size=1.4) +theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11)) + scale_fill_manual(values=c("#eab676","#76b5c5"))

data1=data1[,-c(1,2)]
data1=data.frame(colSums(data1)) 
data1$ind=rownames(data1)
data1$mite="mite"
colnames(data1)=c("nb","ind","te")

data=rbind(data1,data0)
mean(data[data$te=="ltr",]$nb)
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_LTR_fam_bar.png",res=600,width=15000,height = 8000)
ggplot(data,aes(x=ind,y=nb,fill=reorder(te,-nb))) + geom_bar(col="black",stat="identity",col="black",size=1.4) +  scale_fill_manual(values = c("#76b5c5","#eab676"))+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11))


png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_TIPs_LTR_fam_violin.png",res=600,width=15000,height = 8000)
ggplot(data,aes(x=te,y=nb,fill=te)) + geom_violin(col="black",size=1) +  scale_fill_manual(values = c("#eab676","#76b5c5"))+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11))


setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
data0=read.table(file="all_MITEs_ins_TE_ind2.csv2.bed",header=T)
data0$id=paste0(data0$TE,"_",data0$ins)
data0=data0[,-c(1,2)]
rownames=data0$id
data0=data0[,-c(148)]
data1=data.frame(rowSums(data0)) 
colnames(data1)=c("freq")
data1$freq=data1$freq/147
data1$TE="MITE"
nrow(data1[data1$freq>=0.9,])/nrow(data1)
data=data1

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data0=read.table(file="all_LTRs_ins_TE_ind2.csv2.bed",header=T)
data0$id=paste0(data0$TE,"_",data0$ins)
data0=data0[,-c(1,2)]
rownames=data0$id
data0=data0[,-c(148)]
data1=data.frame(rowSums(data0)) 
colnames(data1)=c("freq")
data1$freq=data1$freq/147
data1$TE="LTR"
nrow(data1[data1$freq>=0.9,])/nrow(data1)
data=rbind(data1,data)
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Nb_ins_MITE_LTR_freq_density.png",res=600,width=8000,height = 4000)
ggplot(data,aes(x=freq,fill=TE))+geom_density()+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5)) +  scale_fill_manual(values = c("#eab676","#76b5c5"))+facet_grid(TE~.,scales="free")


data=read.table(file = "/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_win_genes.bed")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_gene_dist_1.png",res=600,width=8000,height = 4000)
ggplot(data,aes(x=V9))+geom_density(fill="#eab676")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5)) + scale_x_continuous(breaks = seq(-5000,5000,1000),limits=c(-5000,5000))


data=read.table(file = "/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv_win_genes.bed")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/MITE_gene_dist_1.png",res=600,width=8000,height = 4000)
ggplot(data,aes(x=V9))+geom_density(fill="#76b5c5")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5)) + scale_x_continuous(breaks = seq(-5000,5000,1000),limits=c(-5000,5000))

data=read.table(file = "/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_MITEs_ins_TE_ind2.csv_win2_div.bed")
data$id=paste0(data$V1,"_",data$V2,"_",data$V3)
data=data[,c(5,4)]
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/MITE_div_windows_barplot.png",res=600,width=8000,height = 4000)
ggplot(data2[data2$V5>=115,],aes(x=reorder(V4,V5),y=V5)) +geom_bar(stat="identity",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + coord_flip() 

```


```{r}
library("ggplot2")
library("dplyr")
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
data=read.table(file="MITE_nb_copies_per_ind_per_TE.txt")

data1=data %>% group_by(V2) %>% summarise_at(vars(V3), list(name = mean))
#data1=data1[data1$name>=30,]
data2=data[data$V2 %in% data1$V2,]
head(data2)
anot=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
data2=left_join(data2,anot,by=c("V2"="V1"))
data2[is.na(data2$V2.y),]$V2.y<-data2[is.na(data2$V2.y),]$V2

mean(data1$name)


png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/boxplot_MITEs_per_ind_per_family.png",res=800,width=21000,height = 5000)
ggplot(data2,aes(x=reorder(V2.y,-V3),y=V3)) + geom_boxplot(outlier.size=0.01) + theme_bw()  +theme(axis.text.x = element_text(angle = 45, hjust = 1,size=3),axis.text.y = element_text( hjust = 1,size=9.5),axis.title = element_text(size = 18),axis.line=element_line(size=0.75)) + stat_summary(fun.y=mean, geom="point", shape=8, size=1.5, color="red2", fill="red2") + labs(x= "TE family",y= "Number of insertions")  + scale_y_continuous(breaks = seq(0, 2000, by = 500))+theme(axis.line=element_line(size=0.75))
mean(data1$name)

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data=read.table(file="LTR_nb_copies_per_ind_per_TE_int.txt")
data=data  %>% group_by(V4,V5) %>% summarise(V3=n())

data1=data %>% group_by(V5) %>% summarise_at(vars(V3), list(name = mean))
#data1=data1[data1$name>=30,]
data2=data[data$V5 %in% data1$V5,]
mean(data1$name)
head(data2)

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/boxplot_LTRs_per_ind_per_family.png",res=800,width=21000,height = 5000)
ggplot(data2,aes(x=reorder(V5,-V3),y=V3)) + geom_boxplot(outlier.size=0.01) + theme_bw()  +theme(axis.text.x = element_text(angle = 45, hjust = 1,size=3),axis.text.y = element_text( hjust = 1,size=9.5),axis.title = element_text(size = 18),axis.line=element_line(size=0.75)) + stat_summary(fun.y=mean, geom="point", shape=8, size=1.5, color="red2", fill="red2") + labs(x= "TE family",y= "Number of insertions")  + scale_y_continuous(breaks = seq(0, 2000, by = 500))+theme(axis.line=element_line(size=0.75))



data=data[,c(1,3)]
data=data%>% group_by(V1) %>% summarise(Frequency = sum(V3))

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_ins_MITE_ind_barplot.png",res=600,width=8000,height = 4000)
ggplot(data,aes(x=V1,y=Frequency))+geom_bar(stat="identity",col="black",fill="purple4")+theme(axis.line=element_line(size=0.75))+theme_bw()+ theme(axis.text.x = element_text(angle = 90,size=7.5))

data$V3="ind"
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_ins_MITE_ind_violin.png",res=600,width=6000,height = 5000)
ggplot(data,aes(x=V3,y=Frequency))+geom_violin(col="black",fill="purple4")+theme(axis.line=element_line(size=0.75))+theme_bw()

data=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/MITE_family_windows_matrics_count_TEfam.txt",sep=" ")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Nb_TIPs_MITE_fam_violin.png",res=600,width=6000,height = 5000)
data$V3="ind"
ggplot(data,aes(x=V3,y=V2))+geom_boxplot(col="black",fill="purple2")+theme(axis.line=element_line(size=0.75))+theme_bw()


data=read.table(file="allTIPs_pop.bed")
data$freq=data$V5/147
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/TIPs_MITE_freq_density.png",res=600,width=10000,height = 5000)
ggplot(data,aes(x=freq))+geom_histogram(col="black",fill="purple4") +theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=1))+theme_bw() + scale_x_continuous(breaks = seq(0,1,0.1))


```

```{r}
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data=read.table(file="allTIPs_pop_dist_gene.bed")
data=data[,c(1:4,9,10)]
data$V10=data$V10/1000
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/TIPs_LTR_gene_dist.png",res=600,width=10000,height = 5000)
ggplot(data,aes(x=V10))+geom_histogram(col="black",fill="dodgerblue3")+theme(axis.line=element_line(size=0.75))+theme_bw() + scale_x_continuous(breaks = seq(-450,400,50))


data=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/genes_LTR_tips_count.txt")
data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/genes_MITE_tips_count.txt")
gene=rbind(data,data1)
gene=data.frame(unique(gene[,c(1)]))
colnames(gene)="V1"
data=left_join(gene,data,by=c("V1"="V1"))
data=left_join(data,data1,by=c("V1"="V1"))
data[is.na(data)]<-0

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Gene_LTR_MITE_TIPs.png",res=600,width=10000,height = 5000)
ggplot(data,aes(x=V2.x,y=V2.y))+geom_point()+theme(axis.line=element_line(size=0.75))

library(reshape2) 
colnames(data)=c("gene","LTR","MITE")
data1=melt(data,id.vars=c("gene"),measure.vars=c("LTR","MITE"))
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Gene_LTR_MITE_TIPs_boxplot.png",res=600,width=5000,height = 7000)
ggplot(data1,aes(x=variable,y=value))+geom_boxplot()+theme(axis.line=element_line(size=0.75))

fn=read.table(file="/home/lgdp/Desktop/gene_TIP_upst_downst/Fagus_Sylvatica_gene_annot_fn2.bed")
fn=fn[,c(4,5)]
data1=left_join(data,fn,by=c("gene"="V4"))
data1$sum=data1$LTR+data1$MITE
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Gene_LTR_MITE_sum_TIPs_boxplot.png",res=600,width=5000,height = 7000)
ggplot(data1,aes(y=sum))+geom_boxplot()+theme(axis.line=element_line(size=0.75))                                                                                                                                      
write.csv(data1,file="TIPs_genes.csv",row.names=F,col.names = T,quote=F)

```


```{r}

library(karyoploteR)
library(dplyr)


cov=read.table(file="~/Desktop/GWAS_plink/BAM_coverage_100kb/all_trees_cov.100kb.regions.mean.bed")
cov=cov[cov$V4<2,]

gene=read.table(file="/home/lgdp/Desktop/Fagus_sylvatica_v3.1.annot.simple.bed")

#data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Fagus_sylvatica_v3_100kbwindows_allTIPs_pop.bed")
#data2=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Fagus_sylvatica_v3_100kbwindows_allTIPs_pop_div.bed")

data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Fagus_sylvatica_v3_100kbwindows_all_MITEs_ins_TE_ind2.csv.bed")
data2=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Fagus_sylvatica_v3_100kbwindows_all_MITEs_ins_TE_ind2.csv_div.bed")
data_m=left_join(data1,data2,by=c("V1"="V1","V2"="V2","V3"="V3"))


#data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Fagus_sylvatica_v3_100kbwindows_allTIPs_pop.bed")
#data2=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Fagus_sylvatica_v3_100kbwindows_allTIPs_pop_div.bed")
data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Fagus_sylvatica_v3_100kbwindows_all_LTRs_ins_TE_ind2.csv.bed")
data2=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Fagus_sylvatica_v3_100kbwindows_all_LTRs_ins_TE_ind2.csv_div.bed")
data_l=left_join(data1,data2,by=c("V1"="V1","V2"="V2","V3"="V3"))

data_snp=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/Beech_147_complet_23062022.recode.Freq.txt")
data_snp=data_snp[,c(1,2)]
data_snp$V3=data_snp$V2+1

genome=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/BA2_mobilome/30scaff.sizes.genome")
custom.genome <- toGRanges(genome)


gene=toGRanges(gene)
setwd("~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("karyoplot_TIPs_ltr_mite.png",width=9000,height = 14000,res=500)
kp <- plotKaryotype(genome=custom.genome,plot.type = 2,cex=0.7)

kpAddBaseNumbers(kp, tick.dist = 5000000, tick.len = 40, tick.col="gray3", cex=0.6,
                 minor.tick.dist = 500000, minor.tick.len = 25, minor.tick.col = "blue")


#kpDataBackground(kp, data.panel = 1, color="white")
#kpAxis(kp, cex=0.30) #ymin = 0, ymax=12
#kpPlotDensity(kp,data=gene, window.size=100000, r0=0, r1=0.4)


kpDataBackground(kp, r0=0, r1=0.45,data.panel = 1, color="white",)
kpAxis(kp, r0=0, r1=0.45,cex=0.35,side=1)
point.colors2 <- colByValue(data_l$V4.y, colors = c("#eab676","black"), min = 1, max=536)
kpPoints(kp,data=toGRanges(data_l), y=data_l$V4.x,col=point.colors2,pch=16,ymin=1, ymax=15,border=point.colors2, r0=0, r1=0.45,cex=0.7)

kpDataBackground(kp, r0=0.55, r1=1,data.panel = 1 , color="white",labels=NA)
kpAxis(kp, r0=0.55, r1=1,cex=0.35, side = 2)
point.colors3 <- colByValue(data_m$V4.y, colors = c("#76b5c5","black"), min = 1, max=337)
kpPoints(kp,data=toGRanges(data_m), y=data_m$V4.x,col=point.colors3,pch=16,ymin=1, ymax=50,border=point.colors3,r0=0.55, r1=1,cex=0.7)

kpDataBackground(kp,data.panel = 2,r0=0.35, r1=1, color="white")
kpAxis(kp,data.panel = 2, r0=0.35, r1=1,cex=0.35,side=1)
#kpPoints(kp,data=toGRanges(data_snp), y=data_snp$V4,col="black",pch=16,ymin=1, ymax=3928,border=point.colors3,r0=0.55, r1=1,cex=0.7)
kpPlotDensity(kp, toGRanges(data_snp), data.panel = 2, window.size = 100000, r0=0.35, r1=1,col="azure3",cex=0.7)

#kpArrows(kp, chr=data$V1, x0=data$V2,x1=data$V2,y0=0.5,y1=-0.05,length=0.06,data.panel = 2,col="green4")

kpBars(kp, data=toGRanges(cov), x0=cov$V2, x1=cov$V3, y0=0.5,y1=-0.05, col="plum4", border="plum4",r0=0, r1=0.3,data.panel = 2)
legend(x = "bottomright", fill = c("#eab676", "#76b5c5","plum4","azure3"), legend = c("LTR-TIPs", "MITE-TIPs","SNPs","No Coverage"))


write.csv(data_m,file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/data_m.csv",quote=F,row.names = F,col.names = F)
write.csv(data_l,file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/data_l.csv",quote=F,row.names = F,col.names = F)
```


```{r}
library(tidyr)
library(dplyr)

data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Fagus_sylvatica_v3_100kbwindows_all_LTRs_ins_TE_ind2.csv.bed")
data2=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Fagus_sylvatica_v3_100kbwindows_all_MITEs_ins_TE_ind2.csv.bed")
data=data.frame(rbind(data1,data2))
data=unique(data[,c(1:3)])
data=left_join(data,data1,by=c("V1"="V1","V2"="V2","V3"="V3"))
data=left_join(data,data2,by=c("V1"="V1","V2"="V2","V3"="V3"))
colnames(data)=c("scaffold","x","y","LTR","MITE")
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/100kb_TIPs_nb_LTR_MITE.png",res=600,width=9000,height = 7000)
ggplot(data,aes(x=LTR,y=MITE))+geom_point()+theme(axis.line=element_line(size=0.75)) +geom_smooth()+ theme_linedraw()                                           

data[is.na(data)]<-0
data$sum=data$LTR+data$MITE
data$id=paste0(data$scaffold,"_",data$x)
data=data[data$sum>=46,]
write.table(data[,c(1:3)],file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_MITEs_ins_TE_ind2.csv_win.top_MITE_LTR_TIPS_nb_100kbwindow.bed",quote=F,sep="\t",row.names = F,col.names = F)

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/sum_top_MITE_LTR_TIPS_nb_100kbwindows_boxplot.png",res=600,width=9000,height = 7000)
ggplot(data[data$sum>=46,],aes(y=reorder(id,sum),x=sum))+geom_bar(fill="black",stat="identity")+theme(axis.line=element_line(size=0.75)) + theme_linedraw()


png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_TIPS_nb_100kbwindows_boxplot.png",res=600,width=9000,height = 7000)
ggplot(data1,aes(y=V4))+geom_boxplot(fill="royalblue4")+theme(axis.line=element_line(size=0.75)) + theme_linedraw()

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/MITE_TIPS_nb_100kbwindows_density.png",res=600,width=9000,height = 7000)
ggplot(data2,aes(x=V4))+geom_density(fill="purple4")+theme(axis.line=element_line(size=0.75)) + theme_linedraw()

data = data %>%
  pivot_longer(cols = c(LTR, MITE), names_to = "Type", values_to = "Value")
data$Type <- factor(data$Type)

d=data[data$Type=="MITE",]
mean(d$Value)

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_MITE_TIPS_nb_100kbwindows_boxplot.png",res=600,width=9000,height = 7000)
ggplot(data,aes(y=Value,fill=Type))+geom_boxplot()+theme(axis.line=element_line(size=0.75)) + theme_linedraw() +scale_fill_manual(values=c("royalblue4","purple4"))

data3=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/Beech_147_complet_23062022.recode.only30Scaff_freq_100kb_nbSNP.bed")
mean(data3$V4)
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/SNP_nb_100kbwindows_boxplot.png",res=600,width=9000,height = 7000)
ggplot(data3,aes(y=V4))+geom_boxplot(fill="darkorange3")+theme(axis.line=element_line(size=0.75)) + theme_linedraw() 


png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/SNP_nb_100kbwindows_density.png",res=600,width=9000,height = 7000)
ggplot(data3,aes(x=V4))+geom_density(fill="darkorange3")+theme(axis.line=element_line(size=0.75)) + theme_linedraw() 


data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/Fagus_sylvatica_v3_100kbwindows_all_MITEs_ins_TE_ind2.csv_div.bed")
data2=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/Fagus_sylvatica_v3_100kbwindows_all_LTRs_ins_TE_ind2.csv_div.bed")
data=data.frame(rbind(data1,data2))
data=unique(data[,c(1:3)])
data=left_join(data,data1,by=c("V1"="V1","V2"="V2","V3"="V3"))
data=left_join(data,data2,by=c("V1"="V1","V2"="V2","V3"="V3"))
colnames(data)=c("scaffold","x","y","LTR_div","MITE_div")

data=left_join(data,data_1,by=c("scaffold"="scaffold","x"="x","y"="y"))
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/100kb_TIPs_div_LTR_MITE.png",res=600,width=9000,height = 7000)
ggplot(data,aes(x=LTR_div,y=MITE_div))+geom_point()+theme(axis.line=element_line(size=0.75)) +geom_smooth() + theme_linedraw()

data$sum2=data$LTR_div+data$MITE_div

data2 = data %>%
  pivot_longer(cols = c(LTR_div, MITE_div), names_to = "Type", values_to = "Value")
data2$Type <- factor(data2$Type)
d=na.omit(data2[data2$Type=="LTR_div",])
mean(d$Value)
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_MITE_TIPS_div_100kbwindows_violin.png",res=600,width=9000,height = 7000)
ggplot(data2,aes(y=Value,x=Type,fill=Type))+geom_violin()+theme(axis.line=element_line(size=0.75)) + theme_linedraw() +scale_fill_manual(values=c("royalblue4","purple4"))

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/100kb_TIPs_div_nb_LTR.png",res=600,width=9000,height = 7000)
ggplot(data,aes(x=LTR_div,y=LTR))+geom_point()+theme(axis.line=element_line(size=0.75)) +geom_smooth() + theme_linedraw()

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/100kb_TIPs_div_nb_MITE.png",res=600,width=9000,height = 7000)
ggplot(data,aes(x=MITE_div,y=MITE))+geom_point()+theme(axis.line=element_line(size=0.75)) +geom_smooth() + theme_linedraw()

png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_TIPS_div_nb_100kbwindows.png",res=600,width=9000,height = 7000)
#ggplot(data,aes(y=MITE_div,x=LTR_div))+geom_point()+theme(axis.line=element_line(size=0.75)) + theme_linedraw()
ggplot(data,aes(y=LTR_div,x=LTR))+geom_point(size=3,fill="royalblue4",shape=25,col="royalblue4")+theme(axis.line=element_line(size=0.75)) + theme_linedraw()


png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/MITE_TIPS_div_nb_100kbwindows.png",res=600,width=9000,height = 7000)
#ggplot(data,aes(y=MITE_div,x=LTR_div))+geom_point()+theme(axis.line=element_line(size=0.75)) + theme_linedraw()
ggplot(data,aes(y=MITE_div,x=MITE))+geom_point(size=3,shape=25,fill="purple4",col="purple4")+theme(axis.line=element_line(size=0.75)) + theme_linedraw()


```


```{r}
setwd("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/")
data=read.table(file="/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/Beech_147_complet_23062022.recode.only30Scaff_freq_100kb_nbSNP.bed")
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/SNP_nb_100kbwindows_density.png",res=600,width=9000,height = 7000)
ggplot(data,aes(x=V4))+geom_density(fill="darkorange3")+theme(axis.line=element_line(size=0.75)) + theme_linedraw()
data$id=paste0(data$V1,"_",data$V2,"_",data$V3)
png("/media/lgdp/Seagate Basic/P2/Abirami/Abirami/vcf_SNP/top20_100kb_nb_SNPs.png",res=600,width=15000,height = 8000)
ggplot(data[data$V4>=3253,],aes(x=reorder(id,V4),y=V4))+geom_bar(col="black",fill="darkorange3",stat="identity")+theme_linedraw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45,size=13,vjust = 1, hjust=1))+coord_flip()

```


```{r}
data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv_count_fam_30scaff.bed")
data1$ins="MITE"
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
png("MITE_3kb_windows_fam_boxplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=ins,y=V4))+geom_boxplot(fill="purple4")+ theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1))
data1$id=paste0(data1$V1,"_",data1$V2)
data1=data1[,c(6,4)]
data1=data1[data1$V4>=41,]
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
png("MITE_3kb_windows_fam_barplot.png",res=800,width=10000,height = 5000)
ggplot(data1,aes(x=reorder(id,V4),y=V4)) +geom_bar(stat="identity",col="black",fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_count_fam_30scaff.bed")
data1$ins="LTR"
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("LTR_10kb_windows_fam.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=ins,y=V4))+geom_boxplot(fill="royalblue4")+ theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,size=1))
data1$id=paste0(data1$V1,"_",data1$V2)
data1=data1[,c(6,4)]
data1=data1[data1$V4>=130,]
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("LTR_10kb_windows_fam_barplot.png",res=800,width=10000,height = 5000)
ggplot(data1,aes(x=reorder(id,V4),y=V4)) +geom_bar(stat="identity",col="black",fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()
```

```{r}
data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv_win_genes_30Scaffold.bed")
data=read.table(file="/home/lgdp/Desktop/distribution_sv/Fagus_Sylvatica_gene_annot_fn2.bed")
data1=left_join(data1,data,by=c("V5"="V4"))
data1$id=paste0(data1$V1.x,"_",data1$V2.x)
data1=unique(data1[,c(10,9)])
data1=data.frame(data1)
data1=data1%>% group_by(V5.y) %>% mutate(count = n())
data1=data1[,c(2,3)]
data1=unique(na.omit(data1))
write.table(data1,file="3kb_genes_TIP_fn_asso_count.bed",sep="\t",quote=F,row.names = F,col.names = F)
data1=data1[data1$count>=37,]
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
png("MITE_3kb_windows_fam_genes_barplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=reorder(V5.y,count),y=count)) +geom_bar(stat="identity",col="black",fill="red4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()


data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_win_genes_30Scaffold.bed")
data=read.table(file="/home/lgdp/Desktop/distribution_sv/Fagus_Sylvatica_gene_annot_fn2.bed")
data1=left_join(data1,data,by=c("V5"="V4"))
data1$id=paste0(data1$V1.x,"_",data1$V2.x)
data1=unique(data1[,c(10,9)])
data1=data.frame(data1)
data1=data1%>% group_by(V5.y) %>% mutate(count = n())
data1=data1[,c(2,3)]
data1=unique(na.omit(data1))
write.table(data1,file="10kb_genes_TIP_fn_asso_count.bed",sep="\t",quote=F,row.names = F,col.names = F)
data1=data1[data1$count>=57,]
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("LTR_10kb_windows_fam_genes_barplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=reorder(V5.y,count),y=count)) +geom_bar(stat="identity",col="black",fill="red4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()
```

```{r}
setwd("~/Desktop/TRACKPOSON_test")
data1=read.table(file="Fagus_sylvatica_v3_10kbwindows_genes.bed")
data=read.table(file="/home/lgdp/Desktop/distribution_sv/Fagus_Sylvatica_gene_annot_fn2.bed")
data1=left_join(data1,data,by=c("V4"="V4"))
data1=data1%>% group_by(V5) %>% mutate(count = n())
data1=unique(na.omit(data1[,c(8,9)]))
data1=data1[data1$count>=89,]
png("10kb_windows_genes_barplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=reorder(V5,count),y=count)) +geom_bar(stat="identity",col="black",fill="yellow4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

write.table(data1,file="10kb_genes_fn_asso_count.bed",sep="\t",quote=F,row.names = F,col.names = F)


setwd("~/Desktop/TRACKPOSON_test")
data1=read.table(file="Fagus_sylvatica_v3_3kbwindows_genes.bed")
data=read.table(file="/home/lgdp/Desktop/distribution_sv/Fagus_Sylvatica_gene_annot_fn2.bed")
data1=left_join(data1,data,by=c("V4"="V4"))
data1=data1%>% group_by(V5) %>% mutate(count = n())
data1=unique(na.omit(data1[,c(8,9)]))
data1=data1[data1$count>=163,]
png("3kb_windows_genes_barplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=reorder(V5,count),y=count)) +geom_bar(stat="identity",col="black",fill="yellow4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

write.table(data1,file="3kb_genes_fn_asso_count.bed",sep="\t",quote=F,row.names = F,col.names = F)


```


```{r}
data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE/all_MITEs_ins_TE_ind2.csv_count_fam-no_TIPs_30_scaff_genes.bed")
data=read.table(file="/home/lgdp/Desktop/distribution_sv/Fagus_Sylvatica_gene_annot_fn2.bed")
data1=left_join(data1,data,by=c("V7"="V4"))
data1=data1%>% group_by(V5.y) %>% mutate(count = n())
data1=data1[,c(11,12)]
data1=unique(na.omit(data1))
write.table(data1,file="3kb_noTIPgenes_fn_asso_count.bed",sep="\t",quote=F,row.names = F,col.names = F)
data1=data1[data1$count>=30,]
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
png("MITE_3kb_windows-noTIP_fam_genes_barplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=reorder(V5.y,count),y=count)) +geom_bar(stat="identity",col="black",fill="yellow4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

data1=read.table(file="/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_count_fam-no_TIPs_30_scaff_genes.bed")
data=read.table(file="/home/lgdp/Desktop/distribution_sv/Fagus_Sylvatica_gene_annot_fn2.bed")
data1=left_join(data1,data,by=c("V7"="V4"))
data1=data1%>% group_by(V5.y) %>% mutate(count = n())
data1=data1[,c(11,12)]
data1=unique(na.omit(data1))
write.table(data1,file="10kb_noTIPgenes_fn_asso_count.bed",sep="\t",quote=F,row.names = F,col.names = F)
data1=data1[data1$count>=30,]
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("LTR_3kb_windows-noTIP_fam_genes_barplot.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=reorder(V5.y,count),y=count)) +geom_bar(stat="identity",col="black",fill="yellow4")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

```
```{r}
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
data=read.table(file="all_MITEs_ins_TE_ind2.csv_count_fam.bed")

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("MITE_windows-TIP_histo.png",res=800,width=15000,height = 8000)
ggplot(data,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#76b5c5")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + scale_x_continuous(breaks = seq(0,160,10),limits=c(0,160))

data$id=paste0(data$V1,"_",data$V2)
data=data[,c(5,4)]

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("MITE_windows-TIP_top10.png",res=800,width=15000,height = 8000)
ggplot(data[data$V4>=50,],aes(x=reorder(id,V4),y=V4)) +geom_bar(stat="identity",col="black",fill="#76b5c5")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  +coord_flip()

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data=read.table(file="all_LTRs_ins_TE_ind2.csv_count_fam.bed")

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("LTR_windows-TIP_histo.png",res=800,width=15000,height = 8000)
ggplot(data,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,280,20),limits=c(0,280))

data$id=paste0(data$V1,"_",data$V2)
data=data[,c(5,4)]

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
png("LTR_windows-TIP_top10.png",res=800,width=15000,height = 8000)
ggplot(data[data$V4>=130,],aes(x=reorder(id,V4),y=V4)) +geom_bar(stat="identity",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + coord_flip() 


setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data1=read.table(file="gene_count_all_MITEs_ins_TE_ind2.csv_win.bed")

png("MITEs_gene-TIP_top20.png",res=800,width=15000,height = 8000)
ggplot(data1[data1$V5>=67,],aes(x=reorder(V4,V5),y=V5)) +geom_bar(stat="identity",col="black",fill="#76b5c5")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + coord_flip() 


setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data2=read.table(file="gene_count_all_LTRs_ins_TE_ind2.csv_win.bed")

png("LTRs_gene-TIP_top20.png",res=800,width=15000,height = 8000)
ggplot(data2[data2$V5>=115,],aes(x=reorder(V4,V5),y=V5)) +geom_bar(stat="identity",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + coord_flip() 

data=rbind(data1,data2)
data=data %>% group_by(V1,V2,V3,V4) %>% 
  summarise(C = sum(V5))

png("MITE_LTRs_gene-TIP_top20_3.png",res=800,width=15000,height = 8000)
ggplot(data[data$C>=164,],aes(x=reorder(V4,C),y=C)) +geom_bar(stat="identity",col="black",fill="lightpink")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + coord_flip() 
data2=data[data$C>=10,]
write.table(data[,c(1:3,5,4)],"all_TIPs_sum_genes.bed",sep="\t",row.names=F,col.names=F,quote=F)
write.table(data2[,c(1:3,5,4)],"all_TIPs_sum_genes_TIPs_10.bed",sep="\t",row.names=F,col.names=F,quote=F)

setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test3/MITE")
datam=read.table(file="all_MITEs_ins_TE_ind2.csv_win.bed")
annot=read.table(file="/home/lgdp/Desktop/FsylCur4_last_annotation_data/FsylCur4_denovoLibTEs_cluster.fa.clustr")
data=left_join(datam,annot,by=c("V4"="V1"))
data[is.na(data$V2.y),]$V2.y<-data[is.na(data$V2.y),]$V4
write.table(data[,c(1,2,3,5)],"~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_MITEs_ins_TE_ind2.csv_win1.bed",sep="\t",row.names = F,col.names = F,quote=F)


setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_LTRs_ins_TE_ind2.csv_win_grp.bed")
png("LTR_windows-TIP.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,272,10),lim=c(0,272))

png("LTR_windows-TIP_1.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,31,1),lim=c(0,31))

png("LTR_windows-TIP_2.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#eab676")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,14,1),lim=c(0,14))

write.table(data1[data1$V4<=14,],file="all_LTRs_ins_TE_ind2.csv_win_grp_final.bed",row.names = F,col.names = F,quote=F,sep="\t")
data2=data1[data1$V4>14,]
write.table(unique(data2[,c(1:3),]),file="all_LTRs_ins_TE_ind2.csv_win_grp_remove.bed",row.names = F,col.names = F,quote=F,sep="\t")


setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
data1=read.table(file="~/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/all_MITEs_ins_TE_ind2.csv_win_grp.bed")
png("MITE_windows-TIP.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#76b5c5")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,152,10),lim=c(0,152))

png("MITE_windows-TIP_1.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#76b5c5")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,31,1),lim=c(0,31))

png("MITE_windows-TIP_2.png",res=800,width=15000,height = 8000)
ggplot(data1,aes(x=V4)) +geom_histogram(stat="count",col="black",fill="#76b5c5")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4)) + scale_x_continuous(breaks = seq(0,22,1),lim=c(0,22))

write.table(data1[data1$V4<=22,],file="all_MITEs_ins_TE_ind2.csv_win_grp_final.bed",row.names = F,col.names = F,quote=F,sep="\t")
data2=data1[data1$V4>22,]
write.table(unique(data2[,c(1:3),]),file="all_MITEs_ins_TE_ind2.csv_win_grp_remove.bed",row.names = F,col.names = F,quote=F,sep="\t")
```


```{r}
setwd("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR")
datal=read.table(file="genes_all_LTRs_ins_TE_ind2.csv_win2_div.bed",header=F)
datal=datal[,c(4,8)]
datal=datal %>% group_by(V4) %>% summarise(nb = mean(V8))
datal$te="ltr"

datam=read.table(file="genes_all_MITEs_ins_TE_ind2.csv_win2_div.bed",header=F)
datam=datam[,c(4,8)]
datam=datam %>% group_by(V4) %>% summarise(nb = mean(V8))
datam$te="mite"

data=rbind(datam,datal)
png("/home/lgdp/Desktop/TRACKPOSON_test/TRACKPOSON_test2/LTR/LTR_MITE_TIPS_gene.png",res=600,width=9000,height = 7000)
ggplot(data,aes(y=nb,x=te,fill=te))+geom_boxplot(size=1.4) +theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =90, vjust = 0.5, hjust=1,size=11)) + scale_fill_manual(values=c("#eab676","#76b5c5"))


gene=data.frame(unique(c(datal$V4,datam$V4)))
colnames(gene)="id"
data=left_join(gene,datal,by=c("id"="V4"))
data=left_join(data,datam,by=c("id"="V4"))
data[is.na(data)]<-0
data$C=data$nb.x+data$nb.y

png("MITE_LTRs_gene-TIP_top20_2.png",res=800,width=15000,height = 8000)
ggplot(data[data$C>=26,],aes(x=reorder(id,C),y=C)) +geom_bar(stat="identity",col="black",fill="black")+theme_bw()+theme(axis.line=element_line(size=0.75),axis.text.x = element_text(angle =45, vjust = 0.5, hjust=1,size=4))  + coord_flip() 

mean(data$sum)
data2=data[,c(1,6)]
annot=read.table(file="/home/lgdp/Desktop/Fagus_sylvatica_v3.1.annot.simple.bed")
data2=left_join(annot,data2,by=c("V4"="id"))
data2[is.na(data2)]<-0
write.table(data2,"all_TIPs_sum_genes.bed",sep="\t",row.names=F,col.names=F,quote=F)
write.table(data2,"all_TIPs_sum_genes_TIPs_4.bed",sep="\t",row.names=F,col.names=F,quote=F)


```

